<!DOCTYPE html>





<html lang="zh">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 3.9.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=7.4.0">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=7.4.0">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=7.4.0">
  <link rel="mask-icon" href="/images/logo.svg?v=7.4.0" color="#222">
  <link rel="alternate" href="/atom.xml" title="詹梦涛的生信博客" type="application/atom+xml">

<link rel="stylesheet" href="/css/main.css?v=7.4.0">


<link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css?v=4.7.0">


<script id="hexo-configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Muse',
    version: '7.4.0',
    exturl: false,
    sidebar: {"position":"left","display":"post","offset":12,"onmobile":false},
    copycode: {"enable":false,"show_result":false,"style":null},
    back2top: {"enable":true,"sidebar":false,"scrollpercent":false},
    bookmark: {"enable":false,"color":"#222","save":"auto"},
    fancybox: false,
    mediumzoom: false,
    lazyload: false,
    pangu: false,
    algolia: {
      appID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    },
    localsearch: {"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},
    path: '',
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    translation: {
      copy_button: 'Copiar',
      copy_success: 'Copiado',
      copy_failure: 'Copiar falló'
    },
    sidebarPadding: 40
  };
</script>

  <meta name="description" content="目前对于GO功能注释的思路有 以下常见的三种： BLAST+InterProScan =&amp;gt; omicsbox可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路： 缩小数">
<meta property="og:type" content="article">
<meta property="og:title" content="De novo 基因组 GO 注释小窗口">
<meta property="og:url" content="http://www.zhanmengtao.top/2019/09/16/GO-annotation/index.html">
<meta property="og:site_name" content="詹梦涛的生信博客">
<meta property="og:description" content="目前对于GO功能注释的思路有 以下常见的三种： BLAST+InterProScan =&amp;gt; omicsbox可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路： 缩小数">
<meta property="og:locale" content="zh">
<meta property="og:updated_time" content="2019-09-17T07:31:02.956Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="De novo 基因组 GO 注释小窗口">
<meta name="twitter:description" content="目前对于GO功能注释的思路有 以下常见的三种： BLAST+InterProScan =&amp;gt; omicsbox可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路： 缩小数">
  <link rel="canonical" href="http://www.zhanmengtao.top/2019/09/16/GO-annotation/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome: false,
    isPost: true,
    isPage: false,
    isArchive: false
  };
</script>

  <title>De novo 基因组 GO 注释小窗口 | 詹梦涛的生信博客</title>
  








  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .logo,
  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh">
  <div class="container use-motion">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-meta">

    <div>
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">詹梦涛的生信博客</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
  </div>

  <div class="site-nav-toggle">
    <button aria-label="Cambiar a barra de navegación">
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>


<nav class="site-nav">
  
  <ul id="menu" class="menu">
      
      
      
        
        <li class="menu-item menu-item-home">
      
    

    <a href="/" rel="section"><i class="menu-item-icon fa fa-fw fa-home"></i> <br>Inicio</a>

  </li>
      
      
      
        
        <li class="menu-item menu-item-archives">
      
    

    <a href="/archives/" rel="section"><i class="menu-item-icon fa fa-fw fa-archive"></i> <br>Archivo</a>

  </li>
  </ul>

    

</nav>
</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
      <article itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block post">
    <link itemprop="mainEntityOfPage" href="http://www.zhanmengtao.top/2019/09/16/GO-annotation/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="詹梦涛">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/images/avatar.gif">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="詹梦涛的生信博客">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">De novo 基因组 GO 注释小窗口

          
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              <span class="post-meta-item-text">Publicado el</span>

              
                
              

              <time title="Creado por: 2019-09-16 20:32:41" itemprop="dateCreated datePublished" datetime="2019-09-16T20:32:41+08:00">2019-09-16</time>
            </span>
          
            

            
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="fa fa-calendar-check-o"></i>
                </span>
                <span class="post-meta-item-text">Editado el</span>
                <time title="Modificado por: 2019-09-17 15:31:02" itemprop="dateModified" datetime="2019-09-17T15:31:02+08:00">2019-09-17</time>
              </span>
            
          

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>目前对于GO功能注释的思路有 以下常见的三种：</p>
<h2 id="BLAST-InterProScan-gt-omicsbox"><a href="#BLAST-InterProScan-gt-omicsbox" class="headerlink" title="BLAST+InterProScan =&gt; omicsbox"></a>BLAST+InterProScan =&gt; omicsbox</h2><p>可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路：</p>
<h3 id="缩小数据库："><a href="#缩小数据库：" class="headerlink" title="缩小数据库："></a>缩小数据库：</h3><p>一种是使用Swiss-Prot等小的数据库进行注释；另一种就是根据基于taxid构建 nr子数据库了，具体的构建可见 基于taxid构建Blast database_bioinfomatics2medicine_新浪博客 </p>
<h3 id="diamond-Accelerated-BLAST-compatible-local-sequence-aligner"><a href="#diamond-Accelerated-BLAST-compatible-local-sequence-aligner" class="headerlink" title="diamond: Accelerated BLAST compatible local sequence aligner."></a>diamond: Accelerated BLAST compatible local sequence aligner.</h3><p>其主页介绍如下：</p>
<p>DIAMOND is a sequence aligner for protein and translated DNA searches, designed for high performance analysis of big sequence data. The key features are:</p>
<p>Pairwise alignment of proteins and translated DNA at 500x-20,000x speed of BLAST.</p>
<p>Frameshift alignments for long read analysis.</p>
<p>Low resource requirements and suitable for running on standard desktops or laptops.</p>
<p>Various output formats, including BLAST pairwise, tabular and XML, as well as taxonomic classification.</p>
<p>它就是那么快，相对于 blastp 的半个月，在 ‘–more-sensitive’ 模式下也仅需要4-5个小时。</p>
<p>在nr注释完之后,就到使用 omicsbox  mapping GO id 了。omicsbox 所有版本均需要购买，但价格比以前（Blast2GO）便宜很多。当然如果条件不允许，那么不如就来个本地化吧。</p>
<h2 id="diamond-及-Blast2go-的联用"><a href="#diamond-及-Blast2go-的联用" class="headerlink" title="diamond 及 Blast2go 的联用"></a>diamond 及 Blast2go 的联用</h2><h3 id="blast2go本地化"><a href="#blast2go本地化" class="headerlink" title="blast2go本地化"></a>blast2go本地化</h3><p>陈连福 centos6.9 代码：</p>
<p>installing Blast2go Databases</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">mkdir /opt/biosoft/blast2go</span><br><span class="line"></span><br><span class="line">cd /opt/biosoft/blast2go</span><br><span class="line"></span><br><span class="line">wget http:<span class="comment">//archive.geneontology.org/latest-full/go_monthly-assocdb-data.gz</span></span><br><span class="line"></span><br><span class="line">ascp -k <span class="number">1</span> -T -l <span class="number">200</span>M -<span class="selector-tag">i</span> ~/.aspera/connect/etc/asperaweb_id_dsa<span class="selector-class">.openssh</span> --host=ftp-private<span class="selector-class">.ncbi</span><span class="selector-class">.nlm</span><span class="selector-class">.nih</span><span class="selector-class">.gov</span> --user=anonftp --mode=recv /gene/DATA/gene_info<span class="selector-class">.gz</span> 。/</span><br><span class="line"></span><br><span class="line">ascp -k <span class="number">1</span> -T -l <span class="number">200</span>M -<span class="selector-tag">i</span> ~/.aspera/connect/etc/asperaweb_id_dsa<span class="selector-class">.openssh</span> --host=ftp-private<span class="selector-class">.ncbi</span><span class="selector-class">.nlm</span><span class="selector-class">.nih</span><span class="selector-class">.gov</span> --user=anonftp --mode=recv /gene/DATA/gene2accession<span class="selector-class">.gz</span> ./</span><br><span class="line"></span><br><span class="line">wget ftp:<span class="comment">//ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb.gz</span></span><br></pre></td></tr></table></figure>

<p>注：对于不能使用aspera加速下载的文件可以试试 lftp -e ‘pget -n NUM -c url; exit’实行多线程下载</p>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">gzip -dv go_monthly-assocdb-data.gz</span><br><span class="line"></span><br><span class="line">gzip -dv gene_info.gz</span><br><span class="line"></span><br><span class="line">gzip -dv gene2accession.gz</span><br><span class="line"></span><br><span class="line">gzip -dv idmapping<span class="selector-class">.tb</span><span class="selector-class">.gz</span></span><br><span class="line"></span><br><span class="line">tar zxf ~/software/local_b2g_db<span class="selector-class">.tar</span><span class="selector-class">.gz</span></span><br><span class="line"></span><br><span class="line">mv local_b2g_db<span class="comment">/* ./  &amp;&amp; rm -rf local_b2g_db/</span></span><br><span class="line"><span class="comment"></span></span><br><span class="line"><span class="comment">perl -p -i -e 's/go_201512-assocdb-data/go_2017-assocdb-data/' install_blast2goDB.sh</span></span><br><span class="line"><span class="comment"></span></span><br><span class="line"><span class="comment">./install_blast2goDB.sh</span></span><br></pre></td></tr></table></figure>

<h3 id="下载NR-数据库到自己的Blast-db-中"><a href="#下载NR-数据库到自己的Blast-db-中" class="headerlink" title="下载NR 数据库到自己的Blast+ db 中"></a>下载NR 数据库到自己的Blast+ db 中</h3><figure class="highlight vim"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/<span class="keyword">blast</span>/executables/<span class="keyword">blast</span>+/<span class="number">2.2</span>.<span class="number">28</span>/ncbi-<span class="keyword">blast</span>-<span class="number">2.2</span>.<span class="number">28</span>+-x64-linux.tar.gz</span><br><span class="line"></span><br><span class="line">tar -zxf ncbi-<span class="keyword">blast</span>-<span class="number">2.2</span>.<span class="number">28</span>+-x64-linux.tar.gz -C /<span class="keyword">opt</span>/biosoft/</span><br><span class="line"></span><br><span class="line">/<span class="keyword">opt</span>/biosoft/ncbi-<span class="keyword">blast</span>-<span class="number">2.2</span>.<span class="number">28</span>+/bin/blastdbcmd -db nr -entry <span class="keyword">all</span> -out nr.faa</span><br></pre></td></tr></table></figure>

<p>值得注意的是这里我用的blastdbcmd是ncbi-blast-2.2.28+的，输出的 nr.faa 文件中序列名是 “&gt;gi|66816243|ref|XP_642131.1| hypothetical protein DDB_G0277827” ，但如果用的是 ncbi-blast-2.6.0+的 blastdbcmd，输出没有GI号，原因可能是：As of September 2016, the integer sequence identifiers known as “GIs” will no longer be included in the GenBank, GenPept, and FASTA formats supported by NCBI for the display of sequence records.In addition, the FASTA format will no longer include the database source abbreviation. Please refer to the NCBI News Announcement posting for more detail.</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget https:<span class="regexp">//gi</span>thub.com<span class="regexp">/bbuchfink/</span>diamond<span class="regexp">/releases/</span>download<span class="regexp">/v0.9.24/</span>diamond-linux64.tar.gz</span><br></pre></td></tr></table></figure>

<p>注：如果在将diamond输出的XML文件导入到 b2g4pipe_v2.5 中出现 “Annotation of 0 seqs with 0 annots finished. Now searching for orfan IPRs…” 这个issues，那可能你的diamond版本有点旧了。</p>
<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">tar -zxf diamond-linux64.tar.gz</span><br><span class="line"></span><br><span class="line">mv diamond ~<span class="string">/bin/</span></span><br><span class="line"></span><br><span class="line">cp <span class="string">../interpro/proteins.fasta</span> <span class="string">./</span></span><br><span class="line"></span><br><span class="line">ascp -k 1 -T -l 200M -i ~<span class="string">/.aspera/connect/etc/asperaweb_id_dsa.openssh</span> <span class="params">--host=ftp-private</span>.ncbi.nlm.nih.gov <span class="params">--user=anonftp</span> <span class="params">--mode=recv</span> <span class="string">/pub/taxonomy/accession2taxid/prot.accession2taxid.gz</span> <span class="string">./prot.accession2taxid.gz</span></span><br><span class="line"></span><br><span class="line">ascp -k 1 -T -l 200M -i ~<span class="string">/.aspera/connect/etc/asperaweb_id_dsa.openssh</span> <span class="params">--host=ftp-private</span>.ncbi.nlm.nih.gov <span class="params">--user=anonftp</span> <span class="params">--mode=recv</span> <span class="string">/pub/taxonomy/taxdmp.zip</span> <span class="string">./</span></span><br><span class="line"></span><br><span class="line">unzip taxdmp.zip</span><br><span class="line"></span><br><span class="line"><span class="string">/opt/biosoft/ncbi-blast-2.2.28</span>+<span class="string">/bin/blastdbcmd</span> -db nr -entry all -out nr.faa</span><br><span class="line"></span><br><span class="line">diamond makedb <span class="params">--in</span> nr.faa -d nr <span class="params">--taxonmap</span> prot.accession2taxid.gz <span class="params">--taxonnodes</span> nodes.dmp <span class="comment">#### DIAMOND v0.9.24</span></span><br><span class="line"></span><br><span class="line">diamond blastp <span class="params">--query</span> proteins.fasta <span class="params">--more-sensitive</span> <span class="params">--db</span> nr <span class="params">--evalue</span> 1e-5 <span class="params">--salltitles</span> <span class="params">--threads</span> 64 <span class="params">--outfmt</span> 5 <span class="params">--out</span> nr.xml</span><br><span class="line"></span><br><span class="line">perl -p -e 's/diamond 0.9.24/BLASTP 2.2.26/' nr.xml &gt; nr_new.xml</span><br><span class="line"></span><br><span class="line">perl -p -i -e 's/^Dbacces.dbname=.*<span class="string">/Dbacces.dbname=b2gdb/</span>' b2gPipe.properties</span><br><span class="line"></span><br><span class="line">perl -p -i -e 's/^Dbacces.dbhost=.*<span class="string">/Dbacces.dbhost=127.0.0.1/</span>' b2gPipe.properties</span><br><span class="line"></span><br><span class="line">java -cp *<span class="function">:ext</span>/*: es.blast2go.prog.B2GAnnotPipe -in <span class="string">../nr_new.xml</span> -out go -annot -dat -annex</span><br></pre></td></tr></table></figure>

<h3 id="整合InterProScan的结果可以使用下面的脚本"><a href="#整合InterProScan的结果可以使用下面的脚本" class="headerlink" title="整合InterProScan的结果可以使用下面的脚本"></a>整合InterProScan的结果可以使用下面的脚本</h3><p>merge_interpro_to_go.pl b2g4pipe/go.annot ../interpro/interpro.tsv &gt; go.annot</p>
<h2 id="InterProScan"><a href="#InterProScan" class="headerlink" title="InterProScan"></a>InterProScan</h2><p>对于 InterProScan 可以本地化，也可以用 interProScan5.pl 将序列发送到官方服务器进行注释，资源允许，还是推荐本地化，网络版感觉还是有点慢～一个本地化教程: </p>
<p>InterProScan的使用教程 <a href="http://www.bioinfo-scrounger.com/archives/164" target="_blank" rel="noopener">http://www.bioinfo-scrounger.com/archives/164</a></p>
<h2 id="eggNOG-Mapper-（包括网页版和本地化）"><a href="#eggNOG-Mapper-（包括网页版和本地化）" class="headerlink" title="eggNOG-Mapper （包括网页版和本地化）"></a>eggNOG-Mapper （包括网页版和本地化）</h2><h3 id="网页版"><a href="#网页版" class="headerlink" title="网页版"></a>网页版</h3><p>关于这个注释方法网页版的推荐自己去摸索吧（就是和NCBI提交类似）=&gt; <a href="http://eggnogdb.embl.de/#/app/emapper" target="_blank" rel="noopener">http://eggnogdb.embl.de/#/app/emapper</a></p>
<h3 id="本地化推荐两个教程吧："><a href="#本地化推荐两个教程吧：" class="headerlink" title="本地化推荐两个教程吧："></a>本地化推荐两个教程吧：</h3><p>序列功能注释神器：eggNOG-mapper，KEGG/COG/KOG/GO/BiGG 一网打尽 ： 序列功能注释神器：eggNOG-mapper，KEGG/COG/KOG/GO/BiGG 一网打尽 « Biostack.org =&gt; <a href="http://www.biostack.org/?p=698" target="_blank" rel="noopener">http://www.biostack.org/?p=698</a></p>
<p>应该是最好的eggnog-mapper功能注释教程：应该是最好的eggnog-mapper功能注释教程 - 简书（生信媛）=&gt; <a href="https://www.jianshu.com/p/e646c0fa6443" target="_blank" rel="noopener">https://www.jianshu.com/p/e646c0fa6443</a></p>
<p>个人思考:</p>
<p>对于准确性的个人思考，基于比对，也就是基因相似性的注释方法(blastp,diamond)可能会取决于数据库的大小，因为在omicsbox中有涉及到相似度cut这一步，如果数据库不够大，那么相似度有的可能只为50(在cut值之上)，所以只使用Swiss-Prot等小数据库注释时可能需要提高cut值.</p>
<p>eggnog的注释，存在两种模式diamond和hummer，前者官方推荐为存在接近物种，也就是在直系同源注释中有更好的表现，hummer则反之. 同样的，只要是比对，就会有cut相似度这一说，数据库的大小及深度(涉及的物种)，就会影响结果。其一个特点在于直系同源注释。这点omicsbox中出了filter with Taxid 来补充。</p>
<p>我个人觉得目前还是blast+interpro的准确性会更高一些，当然如果eggnog持续开发,扩大其数据库，以及支持整合diamond和hummer的结果将会成为新的宠儿。</p>
<p>参考资料</p>
<p><a href="http://blog.sina.com.cn/s/blog_16152d7d70102xnc3.html" target="_blank" rel="noopener">http://blog.sina.com.cn/s/blog_16152d7d70102xnc3.html</a></p>
<p><a href="https://github.com/bbuchfink/diamond" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond</a></p>
<p><a href="http://www.jinciwei.cn/a342744.html" target="_blank" rel="noopener">http://www.jinciwei.cn/a342744.html</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/79" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/79</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/159" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/159</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/165" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/165</a></p>
<p><a href="http://www.biostack.org/?p=698" target="_blank" rel="noopener">http://www.biostack.org/?p=698</a></p>
<p><a href="https://www.jianshu.com/p/e646c0fa6443" target="_blank" rel="noopener">https://www.jianshu.com/p/e646c0fa6443</a></p>
<p>NGS 生物信息学分析 V6.0 陈连福 郑越</p>

    </div>

    
    
    
        
      

      <footer class="post-footer">

        

          <div class="post-nav">
            <div class="post-nav-next post-nav-item">
              
                <a href="/2019/09/16/hello-world/" rel="next" title="Hello World">
                  <i class="fa fa-chevron-left"></i> Hello World
                </a>
              
            </div>

            <span class="post-nav-divider"></span>

            <div class="post-nav-prev post-nav-item">
              
            </div>
          </div>
        
      </footer>
    
  </div>
  
  
  
  </article>

  </div>


          </div>
          

        </div>
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">
        
        
        
        
      

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Tabla de contenidos
        </li>
        <li class="sidebar-nav-overview">
          Inicio
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#BLAST-InterProScan-gt-omicsbox"><span class="nav-number">1.</span> <span class="nav-text">BLAST+InterProScan =&gt; omicsbox</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#缩小数据库："><span class="nav-number">1.1.</span> <span class="nav-text">缩小数据库：</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#diamond-Accelerated-BLAST-compatible-local-sequence-aligner"><span class="nav-number">1.2.</span> <span class="nav-text">diamond: Accelerated BLAST compatible local sequence aligner.</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#diamond-及-Blast2go-的联用"><span class="nav-number">2.</span> <span class="nav-text">diamond 及 Blast2go 的联用</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#blast2go本地化"><span class="nav-number">2.1.</span> <span class="nav-text">blast2go本地化</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#下载NR-数据库到自己的Blast-db-中"><span class="nav-number">2.2.</span> <span class="nav-text">下载NR 数据库到自己的Blast+ db 中</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#整合InterProScan的结果可以使用下面的脚本"><span class="nav-number">2.3.</span> <span class="nav-text">整合InterProScan的结果可以使用下面的脚本</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#InterProScan"><span class="nav-number">3.</span> <span class="nav-text">InterProScan</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#eggNOG-Mapper-（包括网页版和本地化）"><span class="nav-number">4.</span> <span class="nav-text">eggNOG-Mapper （包括网页版和本地化）</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#网页版"><span class="nav-number">4.1.</span> <span class="nav-text">网页版</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#本地化推荐两个教程吧："><span class="nav-number">4.2.</span> <span class="nav-text">本地化推荐两个教程吧：</span></a></li></ol></li></ol></div>
        
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">詹梦涛</p>
  <div class="site-description" itemprop="description"></div>
</div>
  <nav class="site-state motion-element">
      <div class="site-state-item site-state-posts">
        
          <a href="/archives/">
        
          <span class="site-state-item-count">2</span>
          <span class="site-state-item-name">entradas</span>
        </a>
      </div>
    
  </nav>
  <div class="feed-link motion-element">
    <a href="/atom.xml" rel="alternate">
      <i class="fa fa-rss"></i>RSS
    </a>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2019</span>
  <span class="with-love" id="animate">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">詹梦涛</span>
</div>
  <div class="powered-by">Creado mediante <a href="https://hexo.io" class="theme-link" rel="noopener" target="_blank">Hexo</a> v3.9.0</div>
  <span class="post-meta-divider">|</span>
  <div class="theme-info">Tema – <a href="https://theme-next.org" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a> v7.4.0</div>

        












        
      </div>
    </footer>
  </div>

  


  <script src="/lib/anime.min.js?v=3.1.0"></script>
  <script src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  <script src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
<script src="/js/utils.js?v=7.4.0"></script><script src="/js/motion.js?v=7.4.0"></script>
<script src="/js/schemes/muse.js?v=7.4.0"></script>
<script src="/js/next-boot.js?v=7.4.0"></script>



  





















  

  

  

</body>
</html>
