<!DOCTYPE html>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <title>De novo 基因组 GO 注释小窗口 | 詹梦涛的生信博客</title>
  <meta name="keywords" content>
  <meta name="description" content="De novo 基因组 GO 注释小窗口 | 詹梦涛的生信博客">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="description" content="目前对于GO功能注释的思路有 以下常见的三种： 1、BLAST+InterProScan =&amp;gt; omicsbox可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路： (">
<meta property="og:type" content="article">
<meta property="og:title" content="De novo 基因组 GO 注释小窗口">
<meta property="og:url" content="http://yoursite.com/2019/09/16/GO-annotation/index.html">
<meta property="og:site_name" content="詹梦涛的生信博客">
<meta property="og:description" content="目前对于GO功能注释的思路有 以下常见的三种： 1、BLAST+InterProScan =&amp;gt; omicsbox可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路： (">
<meta property="og:locale" content="zh">
<meta property="og:updated_time" content="2019-09-16T13:02:32.176Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="De novo 基因组 GO 注释小窗口">
<meta name="twitter:description" content="目前对于GO功能注释的思路有 以下常见的三种： 1、BLAST+InterProScan =&amp;gt; omicsbox可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路： (">


<link rel="icon" href="/img/avatar.jpg">

<link href="/css/style.css?v=1.0.1" rel="stylesheet">

<link href="/css/hl_theme/atom-light.css?v=1.0.1" rel="stylesheet">

<link href="//cdn.bootcss.com/animate.css/3.5.2/animate.min.css" rel="stylesheet">
<link href="//cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">

<script src="//cdn.bootcss.com/jquery/2.2.4/jquery.min.js"></script>
<script src="/js/jquery.autocomplete.min.js?v=1.0.1"></script>

<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script>
    hljs.initHighlightingOnLoad();
</script>

<script src="//cdn.bootcss.com/nprogress/0.2.0/nprogress.min.js"></script>



<script src="//cdn.bootcss.com/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>

<script src="/js/iconfont.js?v=1.0.1"></script>

</head>
<div style="display: none">
  <input class="theme_disqus_on" value="false">
  <input class="theme_preload_comment" value="false">
  <input class="theme_blog_path" value>
</div>

<body>
<aside class="nav">
    <div class="nav-left">
        <a href="/" class="avatar_target">
    <img class="avatar" src="/img/avatar.jpg" />
</a>
<div class="author">
    <span>詹梦涛</span>
</div>

<div class="icon">
    
        
        <a title="rss" href="/atom.xml" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-rss"></use>
                </svg>
            
        </a>
        
    
        
        <a title="github" href="https://github.com/yelog" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-github"></use>
                </svg>
            
        </a>
        
    
        
        <a title="facebook" href="https://www.facebook.com/faker.tops" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-facebook"></use>
                </svg>
            
        </a>
        
    
        
    
        
    
        
        <a title="instagram" href="https://www.facebook.com/faker.tops" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-instagram"></use>
                </svg>
            
        </a>
        
    
        
        <a title="reddit" href="https://www.reddit.com/user/yelog/" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-reddit"></use>
                </svg>
            
        </a>
        
    
        
        <a title="weibo" href="http://weibo.com/u/2307534817" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-weibo"></use>
                </svg>
            
        </a>
        
    
        
        <a title="jianshu" href="https://www.jianshu.com/u/ff56736de7cf" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-jianshu"></use>
                </svg>
            
        </a>
        
    
        
        <a title="zhihu" href="https://www.zhihu.com/people/jaytp/activities" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-zhihu"></use>
                </svg>
            
        </a>
        
    
        
    
        
        <a title="oschina" href="https://my.oschina.net/yelog" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-oschina"></use>
                </svg>
            
        </a>
        
    
        
    
        
        <a title="email" href="mailto:jaytp@qq.com" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-email"></use>
                </svg>
            
        </a>
        
    
        
        <a title="qq" href="http://wpa.qq.com/msgrd?v=3&uin=872336115&site=qq&menu=yes" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-qq"></use>
                </svg>
            
        </a>
        
    
        
        <a title="kugou" href="https://www.kugou.com/" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-kugou"></use>
                </svg>
            
        </a>
        
    
        
        <a title="neteasemusic" href="https://music.163.com/#/user/home?id=88151013" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-neteasemusic"></use>
                </svg>
            
        </a>
        
    
</div>




<ul>
    <li><div class="all active">全部文章<small>(2)</small></div></li>
    
</ul>
<div class="left-bottom">
    <div class="menus">
    
    
    
    
    </div>
    <div><a class="about  hasFriend  site_url"  href="/about">关于</a><a style="width: 50%"  class="friends">友链</a></div>
</div>
<input type="hidden" id="yelog_site_posts_number" value="2">

<div style="display: none">
    <span id="busuanzi_value_site_uv"></span>
    <span id="busuanzi_value_site_pv"></span>
</div>

    </div>
    <div class="nav-right">
        <div class="friends-area">
    <div class="friends-title">
        友情链接
        <i class="back-title-list"></i>
    </div>
    <div class="friends-content">
        <ul>
            
            <li><a target="_blank" href="http://yelog.org/">叶落阁</a></li>
            
        </ul>
    </div>
</div>
        <div class="title-list">
    <form onkeydown="if(event.keyCode==13){return false;}">
        <input class="search" type="text" placeholder="Search..." autocomplete="off"id="local-search-input" >
        <i class="cross"></i>
        <span>
            <label for="tagswitch">Tags:</label>
            <input id="tagswitch" type="checkbox" style="display: none" />
            <i id="tagsWitchIcon"></i>
        </span>
    </form>
    <div class="tags-list">
    
    <div class="clearfix"></div>
</div>

    
    <nav id="title-list-nav">
        
        <a  class=""
           href="/2019/09/16/hello-world/"
           data-tag=""
           data-author="" >
            <span class="post-title" title="Hello World">Hello World</span>
            <span class="post-date" title="2019-09-16 11:35:12">2019/09/16</span>
        </a>
        
        <a  class=""
           href="/2019/09/16/GO-annotation/"
           data-tag=""
           data-author="" >
            <span class="post-title" title="De novo 基因组 GO 注释小窗口">De novo 基因组 GO 注释小窗口</span>
            <span class="post-date" title="2019-09-16 20:32:41">2019/09/16</span>
        </a>
        
    </nav>
</div>
    </div>
    <div class="hide-list">
        <div class="semicircle">
            <div class="brackets first"><</div>
            <div class="brackets">&gt;</div>
        </div>
    </div>
</aside>
<div class="post">
    <div class="pjax">
        <article id="post-GO-annotation" class="article article-type-post" itemscope itemprop="blogPost">
    
        <h1 class="article-title">De novo 基因组 GO 注释小窗口</h1>
    
    <div class="article-meta">
        
        
        
        
    </div>
    <div class="article-meta">
        
        创建时间:<time class="date" title='更新时间: 2019-09-16 21:02:32'>2019-09-16 20:32</time>
        
    </div>
    <div class="article-meta">
        
        
        <span id="busuanzi_container_page_pv">
            阅读:<span id="busuanzi_value_page_pv">
                <span class="count-comment">
                    <span class="spinner">
                      <div class="cube1"></div>
                      <div class="cube2"></div>
                    </span>
                </span>
            </span>
        </span>
        
        
    </div>
    
    <div class="toc-ref">
    
        <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#1、BLAST-InterProScan-gt-omicsbox"><span class="toc-text">1、BLAST+InterProScan =&gt; omicsbox</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-缩小数据库："><span class="toc-text">(1) 缩小数据库：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-diamond-Accelerated-BLAST-compatible-local-sequence-aligner"><span class="toc-text">(2) diamond: Accelerated BLAST compatible local sequence aligner.</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#diamond-及-Blast2go-的联用"><span class="toc-text">diamond 及 Blast2go 的联用</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-blast2go本地化"><span class="toc-text">(1) blast2go本地化</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-下载NR-数据库到自己的Blast-db-中"><span class="toc-text">(2) 下载NR 数据库到自己的Blast+ db 中</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-整合InterProScan的结果可以使用下面的脚本"><span class="toc-text">(3) 整合InterProScan的结果可以使用下面的脚本</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2、InterProScan"><span class="toc-text">2、InterProScan</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3、eggNOG-Mapper-（包括网页版和本地化）"><span class="toc-text">3、eggNOG-Mapper （包括网页版和本地化）</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-网页版"><span class="toc-text">(1) 网页版</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-本地化推荐两个教程吧："><span class="toc-text">(2) 本地化推荐两个教程吧：</span></a></li></ol></li></ol>
    
<style>
    .left-col .switch-btn,
    .left-col .switch-area {
        display: none;
    }
    .toc-level-3 i,
    .toc-level-3 ol {
        display: none !important;
    }
</style>
</div>
    <div class="article-entry" itemprop="articleBody">
      
        <p>目前对于GO功能注释的思路有 以下常见的三种：</p>
<h2 id="1、BLAST-InterProScan-gt-omicsbox"><a href="#1、BLAST-InterProScan-gt-omicsbox" class="headerlink" title="1、BLAST+InterProScan =&gt; omicsbox"></a>1、BLAST+InterProScan =&gt; omicsbox</h2><p>可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路：</p>
<h3 id="1-缩小数据库："><a href="#1-缩小数据库：" class="headerlink" title="(1) 缩小数据库："></a>(1) 缩小数据库：</h3><p>一种是使用Swiss-Prot等小的数据库进行注释；另一种就是根据基于taxid构建 nr子数据库了，具体的构建可见 基于taxid构建Blast database_bioinfomatics2medicine_新浪博客 </p>
<h3 id="2-diamond-Accelerated-BLAST-compatible-local-sequence-aligner"><a href="#2-diamond-Accelerated-BLAST-compatible-local-sequence-aligner" class="headerlink" title="(2) diamond: Accelerated BLAST compatible local sequence aligner."></a>(2) diamond: Accelerated BLAST compatible local sequence aligner.</h3><p>其主页介绍如下：</p>
<p>DIAMOND is a sequence aligner for protein and translated DNA searches, designed for high performance analysis of big sequence data. The key features are:</p>
<p>Pairwise alignment of proteins and translated DNA at 500x-20,000x speed of BLAST.</p>
<p>Frameshift alignments for long read analysis.</p>
<p>Low resource requirements and suitable for running on standard desktops or laptops.</p>
<p>Various output formats, including BLAST pairwise, tabular and XML, as well as taxonomic classification.</p>
<p>它就是那么快，相对于 blastp 的半个月，在 ‘–more-sensitive’ 模式下也仅需要4-5个小时。</p>
<p>在nr注释完之后,就到使用 omicsbox  mapping GO id 了。omicsbox 所有版本均需要购买，但价格比以前（Blast2GO）便宜很多。当然如果条件不允许，那么不如就来个本地化吧。</p>
<h2 id="diamond-及-Blast2go-的联用"><a href="#diamond-及-Blast2go-的联用" class="headerlink" title="diamond 及 Blast2go 的联用"></a>diamond 及 Blast2go 的联用</h2><h3 id="1-blast2go本地化"><a href="#1-blast2go本地化" class="headerlink" title="(1) blast2go本地化"></a>(1) blast2go本地化</h3><p>陈连福 centos6.9 代码：</p>
<p>installing Blast2go Databases</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">mkdir /opt/biosoft/blast2go</span><br><span class="line"></span><br><span class="line">cd /opt/biosoft/blast2go</span><br><span class="line"></span><br><span class="line">wget http://archive.geneontology.org/latest-full/go_monthly-assocdb-data.gz</span><br><span class="line"></span><br><span class="line">ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /gene/DATA/gene_info.gz 。/</span><br><span class="line"></span><br><span class="line">ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /gene/DATA/gene2accession.gz ./</span><br><span class="line"></span><br><span class="line">wget ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb.gz</span><br></pre></td></tr></table></figure>

<p>注：对于不能使用aspera加速下载的文件可以试试 lftp -e ‘pget -n NUM -c url; exit’实行多线程下载</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">gzip -dv go_monthly-assocdb-data.gz</span><br><span class="line"></span><br><span class="line">gzip -dv gene_info.gz</span><br><span class="line"></span><br><span class="line">gzip -dv gene2accession.gz</span><br><span class="line"></span><br><span class="line">gzip -dv idmapping.tb.gz</span><br><span class="line"></span><br><span class="line">tar zxf ~/software/local_b2g_db.tar.gz</span><br><span class="line"></span><br><span class="line">mv local_b2g_db/* ./  &amp;&amp; rm -rf local_b2g_db/</span><br><span class="line"></span><br><span class="line">perl -p -i -e &apos;s/go_201512-assocdb-data/go_2017-assocdb-data/&apos; install_blast2goDB.sh</span><br><span class="line"></span><br><span class="line">./install_blast2goDB.sh</span><br></pre></td></tr></table></figure>

<h3 id="2-下载NR-数据库到自己的Blast-db-中"><a href="#2-下载NR-数据库到自己的Blast-db-中" class="headerlink" title="(2) 下载NR 数据库到自己的Blast+ db 中"></a>(2) 下载NR 数据库到自己的Blast+ db 中</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.28/ncbi-blast-2.2.28+-x64-linux.tar.gz</span><br><span class="line"></span><br><span class="line">tar -zxf ncbi-blast-2.2.28+-x64-linux.tar.gz -C /opt/biosoft/</span><br><span class="line"></span><br><span class="line">/opt/biosoft/ncbi-blast-2.2.28+/bin/blastdbcmd -db nr -entry all -out nr.faa</span><br></pre></td></tr></table></figure>

<p>值得注意的是这里我用的blastdbcmd是ncbi-blast-2.2.28+的，输出的 nr.faa 文件中序列名是 “&gt;gi|66816243|ref|XP_642131.1| hypothetical protein DDB_G0277827” ，但如果用的是 ncbi-blast-2.6.0+的 blastdbcmd，输出没有GI号，原因可能是：As of September 2016, the integer sequence identifiers known as “GIs” will no longer be included in the GenBank, GenPept, and FASTA formats supported by NCBI for the display of sequence records.In addition, the FASTA format will no longer include the database source abbreviation. Please refer to the NCBI News Announcement posting for more detail.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/bbuchfink/diamond/releases/download/v0.9.24/diamond-linux64.tar.gz</span><br></pre></td></tr></table></figure>

<p>注：如果在将diamond输出的XML文件导入到 b2g4pipe_v2.5 中出现 “Annotation of 0 seqs with 0 annots finished. Now searching for orfan IPRs…” 这个issues，那可能你的diamond版本有点旧了。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">tar -zxf diamond-linux64.tar.gz</span><br><span class="line"></span><br><span class="line">mv diamond ~/bin/</span><br><span class="line"></span><br><span class="line">cp ../interpro/proteins.fasta ./</span><br><span class="line"></span><br><span class="line">ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /pub/taxonomy/accession2taxid/prot.accession2taxid.gz ./prot.accession2taxid.gz</span><br><span class="line"></span><br><span class="line">ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /pub/taxonomy/taxdmp.zip ./</span><br><span class="line"></span><br><span class="line">unzip taxdmp.zip</span><br><span class="line"></span><br><span class="line">/opt/biosoft/ncbi-blast-2.2.28+/bin/blastdbcmd -db nr -entry all -out nr.faa</span><br><span class="line"></span><br><span class="line">diamond makedb --in nr.faa -d nr --taxonmap prot.accession2taxid.gz --taxonnodes nodes.dmp #### DIAMOND v0.9.24</span><br><span class="line"></span><br><span class="line">diamond blastp --query proteins.fasta --more-sensitive --db nr --evalue 1e-5 --salltitles --threads 64 --outfmt 5 --out nr.xml</span><br><span class="line"></span><br><span class="line">perl -p -e &apos;s/diamond 0.9.24/BLASTP 2.2.26/&apos; nr.xml &gt; nr_new.xml</span><br><span class="line"></span><br><span class="line">perl -p -i -e &apos;s/^Dbacces.dbname=.*/Dbacces.dbname=b2gdb/&apos; b2gPipe.properties</span><br><span class="line"></span><br><span class="line">perl -p -i -e &apos;s/^Dbacces.dbhost=.*/Dbacces.dbhost=127.0.0.1/&apos; b2gPipe.properties</span><br><span class="line"></span><br><span class="line">java -cp *:ext/*: es.blast2go.prog.B2GAnnotPipe -in ../nr_new.xml -out go -annot -dat -annex</span><br></pre></td></tr></table></figure>

<h3 id="3-整合InterProScan的结果可以使用下面的脚本"><a href="#3-整合InterProScan的结果可以使用下面的脚本" class="headerlink" title="(3) 整合InterProScan的结果可以使用下面的脚本"></a>(3) 整合InterProScan的结果可以使用下面的脚本</h3><p>merge_interpro_to_go.pl b2g4pipe/go.annot ../interpro/interpro.tsv &gt; go.annot</p>
<h2 id="2、InterProScan"><a href="#2、InterProScan" class="headerlink" title="2、InterProScan"></a>2、InterProScan</h2><p>对于 InterProScan 可以本地化，也可以用 interProScan5.pl 将序列发送到官方服务器进行注释，资源允许，还是推荐本地化，网络版感觉还是有点慢～一个本地化教程: </p>
<p>InterProScan的使用教程 <a href="http://www.bioinfo-scrounger.com/archives/164" target="_blank" rel="noopener">http://www.bioinfo-scrounger.com/archives/164</a></p>
<h2 id="3、eggNOG-Mapper-（包括网页版和本地化）"><a href="#3、eggNOG-Mapper-（包括网页版和本地化）" class="headerlink" title="3、eggNOG-Mapper （包括网页版和本地化）"></a>3、eggNOG-Mapper （包括网页版和本地化）</h2><h3 id="1-网页版"><a href="#1-网页版" class="headerlink" title="(1) 网页版"></a>(1) 网页版</h3><p>关于这个注释方法网页版的推荐自己去摸索吧（就是和NCBI提交类似）=&gt; <a href="http://eggnogdb.embl.de/#/app/emapper" target="_blank" rel="noopener">http://eggnogdb.embl.de/#/app/emapper</a></p>
<h3 id="2-本地化推荐两个教程吧："><a href="#2-本地化推荐两个教程吧：" class="headerlink" title="(2) 本地化推荐两个教程吧："></a>(2) 本地化推荐两个教程吧：</h3><p>序列功能注释神器：eggNOG-mapper，KEGG/COG/KOG/GO/BiGG 一网打尽 ： 序列功能注释神器：eggNOG-mapper，KEGG/COG/KOG/GO/BiGG 一网打尽 « Biostack.org =&gt; <a href="http://www.biostack.org/?p=698" target="_blank" rel="noopener">http://www.biostack.org/?p=698</a></p>
<p>应该是最好的eggnog-mapper功能注释教程：应该是最好的eggnog-mapper功能注释教程 - 简书（生信媛）=&gt; <a href="https://www.jianshu.com/p/e646c0fa6443" target="_blank" rel="noopener">https://www.jianshu.com/p/e646c0fa6443</a></p>
<p>个人思考:</p>
<p>对于准确性的个人思考，基于比对，也就是基因相似性的注释方法(blastp,diamond)可能会取决于数据库的大小，因为在omicsbox中有涉及到相似度cut这一步，如果数据库不够大，那么相似度有的可能只为50(在cut值之上)，所以只使用Swiss-Prot等小数据库注释时可能需要提高cut值.</p>
<p>eggnog的注释，存在两种模式diamond和hummer，前者官方推荐为存在接近物种，也就是在直系同源注释中有更好的表现，hummer则反之. 同样的，只要是比对，就会有cut相似度这一说，数据库的大小及深度(涉及的物种)，就会影响结果。其一个特点在于直系同源注释。这点omicsbox中出了filter with Taxid 来补充。</p>
<p>我个人觉得目前还是blast+interpro的准确性会更高一些，当然如果eggnog持续开发,扩大其数据库，以及支持整合diamond和hummer的结果将会成为新的宠儿。</p>
<p>参考资料</p>
<p><a href="http://blog.sina.com.cn/s/blog_16152d7d70102xnc3.html" target="_blank" rel="noopener">http://blog.sina.com.cn/s/blog_16152d7d70102xnc3.html</a></p>
<p><a href="https://github.com/bbuchfink/diamond" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond</a></p>
<p><a href="http://www.jinciwei.cn/a342744.html" target="_blank" rel="noopener">http://www.jinciwei.cn/a342744.html</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/79" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/79</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/159" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/159</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/165" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/165</a></p>
<p><a href="http://www.biostack.org/?p=698" target="_blank" rel="noopener">http://www.biostack.org/?p=698</a></p>
<p><a href="https://www.jianshu.com/p/e646c0fa6443" target="_blank" rel="noopener">https://www.jianshu.com/p/e646c0fa6443</a></p>
<p>NGS 生物信息学分析 V6.0 陈连福 郑越</p>

      
       <hr><span style="font-style: italic;color: gray;"> 转载请注明来源，欢迎对文章中的引用来源进行考证，欢迎指出任何有错误或不够清晰的表达。可以在下面评论区评论，也可以邮件至 jaytp@qq.com </span>
    </div>
</article>


<p>
    <a href="javascript:void(0)" class="dashang" onclick="dashangToggle()">赏</a>
</p>


<div class="article_copyright">
    <p><span class="copy-title">文章标题:</span>De novo 基因组 GO 注释小窗口</p>
    
    <p><span class="copy-title">本文作者:</span><a href="javascript:void(0)" title="詹梦涛">詹梦涛</a></p>
    <p><span class="copy-title">发布时间:</span>2019-09-16, 20:32:41</p>
    <p><span class="copy-title">最后更新:</span>2019-09-16, 21:02:32</p>
    <span class="copy-title">原始链接:</span><a class="post-url" href="/2019/09/16/GO-annotation/" title="De novo 基因组 GO 注释小窗口">http://yoursite.com/2019/09/16/GO-annotation/</a>
    <p>
        <span class="copy-title">版权声明:</span><i class="fa fa-creative-commons"></i> <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/" title="CC BY-NC-SA 4.0 International" target = "_blank">"署名-非商用-相同方式共享 4.0"</a> 转载请保留原文链接及作者。
    </p>
</div>





    




    </div>
    <div class="copyright">
        <p class="footer-entry">©2016-2019 Yelog</p>
<p class="footer-entry">Built with <a href="https://hexo.io/" target="_blank">Hexo</a> and <a href="https://github.com/yelog/hexo-theme-3-hexo" target="_blank">3-hexo</a> theme</p>

    </div>
    <div class="full-toc">
        <button class="full"><span class="min "></span></button>
<button class="post-toc-menu"><span class="post-toc-menu-icons"></span></button>
<div class="post-toc"><span class="post-toc-title">目录</span>
    <div class="post-toc-content">

    </div>
</div>
<a class="" id="rocket" href="javascript:void(0)"></a>
    </div>
</div>
<div class="acParent"></div>

<div class="hide_box" onclick="dashangToggle()"></div>
<div class="shang_box">
    <a class="shang_close" href="javascript:void(0)" onclick="dashangToggle()">×</a>
    <div class="shang_tit">
        <p>喜欢就点赞,疼爱就打赏</p>
    </div>
    <div class="shang_payimg">
        <div class="pay_img">
            <img src="/img/alipay.jpg" class="alipay" title="扫码支持">
            <img src="/img/weixin.jpg" class="weixin" title="扫码支持">
        </div>
    </div>
    <div class="shang_payselect">
        <span><label><input type="radio" name="pay" checked value="alipay">支付宝</label></span><span><label><input type="radio" name="pay" value="weixin">微信</label></span>
    </div>
</div>


</body>
<script src="/js/jquery.pjax.js?v=1.0.1" ></script>

<script src="/js/script.js?v=1.0.1" ></script>
<script>
    var img_resize = 'default';
    /*作者、标签的自动补全*/
    $(function () {
        $('.search').AutoComplete({
            'data': [],
            'itemHeight': 20,
            'width': 418
        }).AutoComplete('show');
    })
    function initArticle() {
        /*渲染对应的表格样式*/
        
            $(".post .pjax table").addClass("green_title");
        

        /*渲染打赏样式*/
        
        $("input[name=pay]").on("click", function () {
            if($("input[name=pay]:checked").val()=="weixin"){
                $(".shang_box .shang_payimg .pay_img").addClass("weixin_img");
            } else {
                $(".shang_box .shang_payimg .pay_img").removeClass("weixin_img");
            }
        })
        

        /*高亮代码块行号*/
        
        $('pre code').each(function(){
            var lines = $(this).text().split('\n').length - 1, widther='';
            if (lines>99) {
                widther = 'widther'
            }
            var $numbering = $('<ul/>').addClass('pre-numbering ' + widther).attr("unselectable","on");
            $(this).addClass('has-numbering ' + widther)
                    .parent()
                    .append($numbering);
            for(var i=1;i<=lines;i++){
                $numbering.append($('<li/>').text(i));
            }
        });
        

        /*访问数量*/
        
        $.getScript("//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js");
        

        /*代码高亮，行号对齐*/
        $('.pre-numbering').css('line-height',$('.has-numbering').css('line-height'));

        
    }

    /*打赏页面隐藏与展示*/
    
    function dashangToggle() {
        $(".shang_box").fadeToggle();
        $(".hide_box").fadeToggle();
    }
    

</script>

<!--加入行号的高亮代码块样式-->

<style>
    pre{
        position: relative;
        margin-bottom: 24px;
        border-radius: 10px;
        border: 1px solid #e2dede;
        background: #FFF;
        overflow: hidden;
    }
    code.has-numbering{
        margin-left: 30px;
    }
    code.has-numbering.widther{
        margin-left: 35px;
    }
    .pre-numbering{
        margin: 0px;
        position: absolute;
        top: 0;
        left: 0;
        width: 20px;
        padding: 0.5em 3px 0.7em 5px;
        border-right: 1px solid #C3CCD0;
        text-align: right;
        color: #AAA;
        background-color: #fafafa;
    }
    .pre-numbering.widther {
        width: 35px;
    }
</style>

<!--自定义样式设置-->
<style>
    
    
    .nav {
        width: 542px;
    }
    .nav.fullscreen {
        margin-left: -542px;
    }
    .nav-left {
        width: 120px;
    }
    
    
    @media screen and (max-width: 1468px) {
        .nav {
            width: 492px;
        }
        .nav.fullscreen {
            margin-left: -492px;
        }
        .nav-left {
            width: 100px;
        }
    }
    
    
    @media screen and (max-width: 1024px) {
        .nav {
            width: 492px;
            margin-left: -492px
        }
        .nav.fullscreen {
            margin-left: 0;
        }
        .nav .hide-list.fullscreen {
            left: 492px
        }
    }
    
    @media screen and (max-width: 426px) {
        .nav {
            width: 100%;
        }
        .nav-left {
            width: 100%;
        }
    }
    
    
    .nav-right .title-list nav a .post-title, .nav-right .title-list #local-search-result a .post-title {
        color: #383636;
    }
    
    
    .nav-right .title-list nav a .post-date, .nav-right .title-list #local-search-result a .post-date {
        color: #5e5e5f;
    }
    
    
    .nav-right nav a.hover, #local-search-result a.hover{
        background-color: #e2e0e0;
    }
    
    

    /*列表样式*/
    
    .post .pjax article .article-entry>ol, .post .pjax article .article-entry>ul, .post .pjax article>ol, .post .pjax article>ul{
        border: #e2dede solid 1px;
        border-radius: 10px;
        padding: 10px 32px 10px 56px;
    }
    .post .pjax article .article-entry li>ol, .post .pjax article .article-entry li>ul,.post .pjax article li>ol, .post .pjax article li>ul{
        padding-top: 5px;
        padding-bottom: 5px;
    }
    .post .pjax article .article-entry>ol>li, .post .pjax article .article-entry>ul>li,.post .pjax article>ol>li, .post .pjax article>ul>li{
        margin-bottom: auto;
        margin-left: auto;
    }
    .post .pjax article .article-entry li>ol>li, .post .pjax article .article-entry li>ul>li,.post .pjax article li>ol>li, .post .pjax article li>ul>li{
        margin-bottom: auto;
        margin-left: auto;
    }
    

    /* 背景图样式 */
    
    


    /*引用块样式*/
    

    /*文章列表背景图*/
    
    .nav-right:before {
        content: ' ';
        display: block;
        position: absolute;
        left: 0;
        top: 0;
        width: 100%;
        height: 100%;
        opacity: 0.3;
        background: url("https://i.loli.net/2019/07/22/5d3521411f3f169375.png");
        background-repeat: no-repeat;
        background-position: 50% 0;
        -ms-background-size: cover;
        -o-background-size: cover;
        -moz-background-size: cover;
        -webkit-background-size: cover;
        background-size: cover;
    }
    

    
</style>







</html>
