<!DOCTYPE html>
<html>
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  <title>詹梦涛的生信博客</title>
  <meta name="keywords" content>
  <meta name="description" content="詹梦涛的生信博客">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="description" content="Welcome to Hexo! This is your very first post. Check documentation for more info. If you get any problems when using Hexo, you can find the answer in troubleshooting or you can ask me on GitHub. Quick">
<meta property="og:type" content="article">
<meta property="og:title" content="Hello World">
<meta property="og:url" content="http://yoursite.com/2019/09/16/hello-world/index.html">
<meta property="og:site_name" content="詹梦涛的生信博客">
<meta property="og:description" content="Welcome to Hexo! This is your very first post. Check documentation for more info. If you get any problems when using Hexo, you can find the answer in troubleshooting or you can ask me on GitHub. Quick">
<meta property="og:locale" content="zh">
<meta property="og:updated_time" content="2019-09-16T03:35:12.763Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Hello World">
<meta name="twitter:description" content="Welcome to Hexo! This is your very first post. Check documentation for more info. If you get any problems when using Hexo, you can find the answer in troubleshooting or you can ask me on GitHub. Quick">


<link rel="icon" href="/img/avatar.jpg">

<link href="/css/style.css?v=1.0.1" rel="stylesheet">

<link href="/css/hl_theme/atom-light.css?v=1.0.1" rel="stylesheet">

<link href="//cdn.bootcss.com/animate.css/3.5.2/animate.min.css" rel="stylesheet">
<link href="//cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">

<script src="//cdn.bootcss.com/jquery/2.2.4/jquery.min.js"></script>
<script src="/js/jquery.autocomplete.min.js?v=1.0.1"></script>

<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script>
    hljs.initHighlightingOnLoad();
</script>

<script src="//cdn.bootcss.com/nprogress/0.2.0/nprogress.min.js"></script>



<script src="//cdn.bootcss.com/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>

<script src="/js/iconfont.js?v=1.0.1"></script>

</head>
<div style="display: none">
  <input class="theme_disqus_on" value="false">
  <input class="theme_preload_comment" value="false">
  <input class="theme_blog_path" value>
</div>

<body>
<aside class="nav">
    <div class="nav-left">
        <a href="/" class="avatar_target">
    <img class="avatar" src="/img/avatar.jpg" />
</a>
<div class="author">
    <span>詹梦涛</span>
</div>

<div class="icon">
    
        
    
        
        <a title="github" href="https://github.com/Zhanmengtao" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-github"></use>
                </svg>
            
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a title="jianshu" href="https://www.jianshu.com/u/f959bbbf2062" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-jianshu"></use>
                </svg>
            
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a title="email" href="mailto:zhanmengtao123@163.com" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-email"></use>
                </svg>
            
        </a>
        
    
        
        <a title="qq" href="http://wpa.qq.com/msgrd?v=3&uin=1406390902&site=qq&menu=yes" target="_blank">
            
                <svg class="iconfont-svg" aria-hidden="true">
                    <use xlink:href="#icon-qq"></use>
                </svg>
            
        </a>
        
    
        
    
        
    
</div>




<ul>
    <li><div class="all active">全部文章<small>(2)</small></div></li>
    
</ul>
<div class="left-bottom">
    <div class="menus">
    
    
    
    
    </div>
    <div><a class="about  hasFriend  site_url"  href="/about">关于</a><a style="width: 50%"  class="friends">友链</a></div>
</div>
<input type="hidden" id="yelog_site_posts_number" value="2">

<div style="display: none">
    <span id="busuanzi_value_site_uv"></span>
    <span id="busuanzi_value_site_pv"></span>
</div>

    </div>
    <div class="nav-right">
        <div class="friends-area">
    <div class="friends-title">
        友情链接
        <i class="back-title-list"></i>
    </div>
    <div class="friends-content">
        <ul>
            
            <li><a target="_blank" href="http://xuzhougeng.top/">徐洲更</a></li>
            
        </ul>
    </div>
</div>
        <div class="title-list">
    <form onkeydown="if(event.keyCode==13){return false;}">
        <input class="search" type="text" placeholder="Search..." autocomplete="off"id="local-search-input" >
        <i class="cross"></i>
        <span>
            <label for="tagswitch">Tags:</label>
            <input id="tagswitch" type="checkbox" style="display: none" />
            <i id="tagsWitchIcon"></i>
        </span>
    </form>
    <div class="tags-list">
    
    <div class="clearfix"></div>
</div>

    
    <nav id="title-list-nav">
        
        <a  class=""
           href="/2019/09/16/hello-world/"
           data-tag=""
           data-author="" >
            <span class="post-title" title="Hello World">Hello World</span>
            <span class="post-date" title="2019-09-16 11:35:12">2019/09/16</span>
        </a>
        
        <a  class=""
           href="/2019/09/16/De-novo-基因组GO-注释小窗口/"
           data-tag=""
           data-author="" >
            <span class="post-title" title=""></span>
            <span class="post-date" title="2019-09-16 17:28:04">2019/09/16</span>
        </a>
        
    </nav>
</div>
    </div>
    <div class="hide-list">
        <div class="semicircle">
            <div class="brackets first"><</div>
            <div class="brackets">&gt;</div>
        </div>
    </div>
</aside>
<div class="post">
    <div class="pjax">
        <article id="post-De-novo-基因组GO-注释小窗口" class="article article-type-post" itemscope itemprop="blogPost">
    
    <div class="article-meta">
        
        
        
        
    </div>
    <div class="article-meta">
        
        创建时间:<time class="date" title='更新时间: 2019-09-16 17:30:05'>2019-09-16 17:28</time>
        
    </div>
    <div class="article-meta">
        
        
        <span id="busuanzi_container_page_pv">
            阅读:<span id="busuanzi_value_page_pv">
                <span class="count-comment">
                    <span class="spinner">
                      <div class="cube1"></div>
                      <div class="cube2"></div>
                    </span>
                </span>
            </span>
        </span>
        
        
    </div>
    
    <div class="toc-ref">
    
        <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#tags"><span class="toc-text">tags:</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#对于不能使用aspera加速下载的文件可以试试-lftp-e-‘pget-n-NUM-c-url-exit’实行多线程下载"><span class="toc-text">对于不能使用aspera加速下载的文件可以试试 lftp -e ‘pget -n NUM -c url; exit’实行多线程下载</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#值得注意的是这里我用的blastdbcmd是ncbi-blast-2-2-28-的，输出的-nr-faa-文件中序列名是-“-gt-gi-66816243-ref-XP-642131-1-hypothetical-protein-DDB-G0277827”-，但如果用的是-ncbi-blast-2-6-0-的-blastdbcmd，输出没有GI号，原因可能是：As-of-September-2016-the-integer-sequence-identifiers-known-as-“GIs”-will-no-longer-be-included-in-the-GenBank-GenPept-and-FASTA-formats-supported-by-NCBI-for-the-display-of-sequence-records-In-addition-the-FASTA-format-will-no-longer-include-the-database-source-abbreviation-Please-refer-to-the-NCBI-News-Announcement-posting-for-more-detail"><span class="toc-text">值得注意的是这里我用的blastdbcmd是ncbi-blast-2.2.28+的，输出的 nr.faa 文件中序列名是 “&gt;gi|66816243|ref|XP_642131.1| hypothetical protein DDB_G0277827” ，但如果用的是 ncbi-blast-2.6.0+的 blastdbcmd，输出没有GI号，原因可能是：As of September 2016, the integer sequence identifiers known as “GIs” will no longer be included in the GenBank, GenPept, and FASTA formats supported by NCBI for the display of sequence records.In addition, the FASTA format will no longer include the database source abbreviation. Please refer to the NCBI News Announcement posting for more detail.</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#同样注意版本，因为diamond输出的XML与-b2g4pipe-v2-5-不兼容而出现-“Annotation-of-0-seqs-with-0-annots-finished-Now-searching-for-orfan-IPRs…”-这个issues的讨论，你可以在到下面的链接去看。"><span class="toc-text">同样注意版本，因为diamond输出的XML与 b2g4pipe_v2.5 不兼容而出现 “Annotation of 0 seqs with 0 annots finished. Now searching for orfan IPRs…” 这个issues的讨论，你可以在到下面的链接去看。</span></a></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#至于整合InterProScan的结果可以使用下面的脚本，当然也可以在java-cp-ext-es-blast2go-prog-B2GAnnotPipe-ips-来实现，或者将b2g4pipe的输出文件加载到-Blast2GO-Basic-中进行后续可视化分析。"><span class="toc-text">至于整合InterProScan的结果可以使用下面的脚本，当然也可以在java -cp :ext/: es.blast2go.prog.B2GAnnotPipe -ips 来实现，或者将b2g4pipe的输出文件加载到 Blast2GO Basic 中进行后续可视化分析。</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Anyone-use-diamond-output-xml-file-for-b2gpipe-·-Issue-79-·-bbuchfink-diamond-·-GitHub"><span class="toc-text">Anyone use diamond output xml file  for b2gpipe? · Issue #79 · bbuchfink/diamond · GitHub</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Problem-in-uploading-the-Diamond-blastx-results-into-Blast2GO-·-Issue-159-·-bbuchfink-diamond-·-GitHub"><span class="toc-text">Problem in uploading the Diamond blastx results into Blast2GO · Issue #159 · bbuchfink/diamond · GitHub</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#diamond-to-blast2go-erro-·-Issue-165-·-bbuchfink-diamond-·-GitHub"><span class="toc-text">diamond to blast2go erro  · Issue #165 · bbuchfink/diamond · GitHub</span></a>
    
<style>
    .left-col .switch-btn,
    .left-col .switch-area {
        display: none;
    }
    .toc-level-3 i,
    .toc-level-3 ol {
        display: none !important;
    }
</style>
</div>
    <div class="article-entry" itemprop="articleBody">
      
        <hr>
<p>title: De novo 基因组GO 注释小窗口<br>date: 2019-09-16 08:48:48</p>
<h2 id="tags"><a href="#tags" class="headerlink" title="tags:"></a>tags:</h2><p>关于GO 注释的心得体会<br>目前对于GO功能注释的思路有 以下常见的三种：<br>1、BLAST+InterProScan =&gt; omicsbox</p>
<p>可以先通过使用 blastp 进行nr注释，那么问题就来了，nr 数据库截至2018年12月已经有120G左右的数据量了，根据我的个人使用经验，对25000条蛋白进行32线程的并行化比对，差不多要半个月左右（心中一万匹cnm飘过）。很显然，在计算资源有限的条件下这是不现实的，那么大概就会出现两种思路：</p>
<p>（1）缩小数据库：一种是使用Swiss-Prot等小的数据库进行注释；另一种就是根据基于taxid构建 nr子数据库了，具体的构建可见 基于taxid构建Blast database_bioinfomatics2medicine_新浪博客 （2）能不能使用一个更高效的比对工具呢，答案是肯定的 那就是 GitHub - bbuchfink/diamond: Accelerated BLAST compatible local sequence aligner.</p>
<p>其主页介绍如下：</p>
<p>DIAMOND is a sequence aligner for protein and translated DNA searches, designed for high performance analysis of big sequence data. The key features are:</p>
<p>Pairwise alignment of proteins and translated DNA at 500x-20,000x speed of BLAST.</p>
<p>Frameshift alignments for long read analysis.</p>
<p>Low resource requirements and suitable for running on standard desktops or laptops.</p>
<p>Various output formats, including BLAST pairwise, tabular and XML, as well as taxonomic classification.</p>
<p>对的，它就是那么快，相对于 blastp 的半个月，在 ‘–more-sensitive’ 模式下也仅需要4-5个小时。</p>
<p>在nr注释完之后,就到使用 Blast2GO mapping GO id 了。omicsbox 所有版本均需要购买，但价格比以前（Blast2GO）便宜很多。当然如果条件不允许，那么不如就来个本地化吧，对于本地化网上也有很多的教程了：</p>
<p>blast2go本地化2017教程：blast2go本地化2017教程 - 生信技能树</p>
<p>非root权限的blast2go的安装和使用（二）· blast2go的数据和软件准备及使用：<a href="https://mp.weixin.qq.com/s?__biz=MzI1MjU5MjMzNA==&amp;mid=2247486358&amp;idx=1&amp;sn=4b095e9a200a419079d947d930c38abd&amp;chksm=e9e02237de97ab21f4227ee76eca0aa94433fe6cc374e030093eb3acc2094a546110763bb411&amp;mpshare=1&amp;scene=23&amp;srcid=0101kMbzD2el0RPASBiC9pFi#rd" target="_blank" rel="noopener">https://mp.weixin.qq.com/s?__biz=MzI1MjU5MjMzNA==&amp;mid=2247486358&amp;idx=1&amp;sn=4b095e9a200a419079d947d930c38abd&amp;chksm=e9e02237de97ab21f4227ee76eca0aa94433fe6cc374e030093eb3acc2094a546110763bb411&amp;mpshare=1&amp;scene=23&amp;srcid=0101kMbzD2el0RPASBiC9pFi#rd</a></p>
<p>陈连福 centos6.9 代码：</p>
<p>“# installing Blast2go Databases</p>
<p>mkdir /opt/biosoft/blast2go</p>
<p>cd /opt/biosoft/blast2go</p>
<p>wget <a href="http://archive.geneontology.org/latest-full/go_monthly-assocdb-data.gz" target="_blank" rel="noopener">http://archive.geneontology.org/latest-full/go_monthly-assocdb-data.gz</a></p>
<p>wget <a href="ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene_info.gz" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene_info.gz</a> ### ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh –host=ftp-private.ncbi.nlm.nih.gov –user=anonftp –mode=recv /gene/DATA/gene_info.gz 。/</p>
<p>wget <a href="ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2accession.gz" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2accession.gz</a>  #### ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh –host=ftp-private.ncbi.nlm.nih.gov –user=anonftp –mode=recv /gene/DATA/gene2accession.gz ./</p>
<p>wget <a href="ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb.gz" target="_blank" rel="noopener">ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb.gz</a></p>
<h2 id="对于不能使用aspera加速下载的文件可以试试-lftp-e-‘pget-n-NUM-c-url-exit’实行多线程下载"><a href="#对于不能使用aspera加速下载的文件可以试试-lftp-e-‘pget-n-NUM-c-url-exit’实行多线程下载" class="headerlink" title="对于不能使用aspera加速下载的文件可以试试 lftp -e ‘pget -n NUM -c url; exit’实行多线程下载"></a>对于不能使用aspera加速下载的文件可以试试 lftp -e ‘pget -n NUM -c url; exit’实行多线程下载</h2><p>gzip -dv go_monthly-assocdb-data.gz</p>
<p>gzip -dv gene_info.gz</p>
<p>gzip -dv gene2accession.gz</p>
<p>gzip -dv idmapping.tb.gz</p>
<p>tar zxf ~/software/local_b2g_db.tar.gz</p>
<p>mv local_b2g_db/* ./  &amp;&amp; rm -rf local_b2g_db/</p>
<p>perl -p -i -e ‘s/go_201512-assocdb-data/go_2017-assocdb-data/‘ install_blast2goDB.sh</p>
<p>./install_blast2goDB.sh</p>
<p>#一个自己的例子 #</p>
<p>##下载NR 数据库到自己的Blast+ db 中</p>
<p>wget <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.28/ncbi-blast-2.2.28+-x64-linux.tar.gz" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.28/ncbi-blast-2.2.28+-x64-linux.tar.gz</a></p>
<p>tar -zxf ncbi-blast-2.2.28+-x64-linux.tar.gz -C /opt/biosoft/</p>
<p>/opt/biosoft/ncbi-blast-2.2.28+/bin/blastdbcmd -db nr -entry all -out nr.faa</p>
<h3 id="值得注意的是这里我用的blastdbcmd是ncbi-blast-2-2-28-的，输出的-nr-faa-文件中序列名是-“-gt-gi-66816243-ref-XP-642131-1-hypothetical-protein-DDB-G0277827”-，但如果用的是-ncbi-blast-2-6-0-的-blastdbcmd，输出没有GI号，原因可能是：As-of-September-2016-the-integer-sequence-identifiers-known-as-“GIs”-will-no-longer-be-included-in-the-GenBank-GenPept-and-FASTA-formats-supported-by-NCBI-for-the-display-of-sequence-records-In-addition-the-FASTA-format-will-no-longer-include-the-database-source-abbreviation-Please-refer-to-the-NCBI-News-Announcement-posting-for-more-detail"><a href="#值得注意的是这里我用的blastdbcmd是ncbi-blast-2-2-28-的，输出的-nr-faa-文件中序列名是-“-gt-gi-66816243-ref-XP-642131-1-hypothetical-protein-DDB-G0277827”-，但如果用的是-ncbi-blast-2-6-0-的-blastdbcmd，输出没有GI号，原因可能是：As-of-September-2016-the-integer-sequence-identifiers-known-as-“GIs”-will-no-longer-be-included-in-the-GenBank-GenPept-and-FASTA-formats-supported-by-NCBI-for-the-display-of-sequence-records-In-addition-the-FASTA-format-will-no-longer-include-the-database-source-abbreviation-Please-refer-to-the-NCBI-News-Announcement-posting-for-more-detail" class="headerlink" title="值得注意的是这里我用的blastdbcmd是ncbi-blast-2.2.28+的，输出的 nr.faa 文件中序列名是 “&gt;gi|66816243|ref|XP_642131.1| hypothetical protein DDB_G0277827” ，但如果用的是 ncbi-blast-2.6.0+的 blastdbcmd，输出没有GI号，原因可能是：As of September 2016, the integer sequence identifiers known as “GIs” will no longer be included in the GenBank, GenPept, and FASTA formats supported by NCBI for the display of sequence records.In addition, the FASTA format will no longer include the database source abbreviation. Please refer to the NCBI News Announcement posting for more detail."></a>值得注意的是这里我用的blastdbcmd是ncbi-blast-2.2.28+的，输出的 nr.faa 文件中序列名是 “&gt;gi|66816243|ref|XP_642131.1| hypothetical protein DDB_G0277827” ，但如果用的是 ncbi-blast-2.6.0+的 blastdbcmd，输出没有GI号，原因可能是：As of September 2016, the integer sequence identifiers known as “GIs” will no longer be included in the GenBank, GenPept, and FASTA formats supported by NCBI for the display of sequence records.In addition, the FASTA format will no longer include the database source abbreviation. Please refer to the NCBI News Announcement posting for more detail.</h3><p>wget <a href="https://github.com/bbuchfink/diamond/releases/download/v0.9.24/diamond-linux64.tar.gz" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/releases/download/v0.9.24/diamond-linux64.tar.gz</a></p>
<h3 id="同样注意版本，因为diamond输出的XML与-b2g4pipe-v2-5-不兼容而出现-“Annotation-of-0-seqs-with-0-annots-finished-Now-searching-for-orfan-IPRs…”-这个issues的讨论，你可以在到下面的链接去看。"><a href="#同样注意版本，因为diamond输出的XML与-b2g4pipe-v2-5-不兼容而出现-“Annotation-of-0-seqs-with-0-annots-finished-Now-searching-for-orfan-IPRs…”-这个issues的讨论，你可以在到下面的链接去看。" class="headerlink" title="同样注意版本，因为diamond输出的XML与 b2g4pipe_v2.5 不兼容而出现 “Annotation of 0 seqs with 0 annots finished. Now searching for orfan IPRs…” 这个issues的讨论，你可以在到下面的链接去看。"></a>同样注意版本，因为diamond输出的XML与 b2g4pipe_v2.5 不兼容而出现 “Annotation of 0 seqs with 0 annots finished. Now searching for orfan IPRs…” 这个issues的讨论，你可以在到下面的链接去看。</h3><p>tar -zxf diamond-linux64.tar.gz</p>
<p>mv diamond ~/bin/</p>
<p>cp ../interpro/proteins.fasta ./</p>
<p>ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh –host=ftp-private.ncbi.nlm.nih.gov –user=anonftp –mode=recv /pub/taxonomy/accession2taxid/prot.accession2taxid.gz ./prot.accession2taxid.gz</p>
<p>ascp -k 1 -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh –host=ftp-private.ncbi.nlm.nih.gov –user=anonftp –mode=recv /pub/taxonomy/taxdmp.zip ./</p>
<p>unzip taxdmp.zip</p>
<p>/opt/biosoft/ncbi-blast-2.2.28+/bin/blastdbcmd -db nr -entry all -out nr.faa</p>
<p>diamond makedb –in nr.faa -d nr –taxonmap prot.accession2taxid.gz –taxonnodes nodes.dmp #### DIAMOND v0.9.24</p>
<p>diamond blastp –query proteins.fasta –more-sensitive –db nr –evalue 1e-5 –salltitles –threads 64 –outfmt 5 –out nr.xml</p>
<p>perl -p -e ‘s/diamond 0.9.24/BLASTP 2.2.26/‘ nr.xml &gt; nr_new.xml</p>
<p>perl -p -i -e ‘s/^Dbacces.dbname=.*/Dbacces.dbname=b2gdb/‘ b2gPipe.properties</p>
<p>perl -p -i -e ‘s/^Dbacces.dbhost=.*/Dbacces.dbhost=127.0.0.1/‘ b2gPipe.properties</p>
<p>java -cp <em>:ext/</em>: es.blast2go.prog.B2GAnnotPipe -in ../nr_new.xml -out go -annot -dat -annex</p>
<h1 id="至于整合InterProScan的结果可以使用下面的脚本，当然也可以在java-cp-ext-es-blast2go-prog-B2GAnnotPipe-ips-来实现，或者将b2g4pipe的输出文件加载到-Blast2GO-Basic-中进行后续可视化分析。"><a href="#至于整合InterProScan的结果可以使用下面的脚本，当然也可以在java-cp-ext-es-blast2go-prog-B2GAnnotPipe-ips-来实现，或者将b2g4pipe的输出文件加载到-Blast2GO-Basic-中进行后续可视化分析。" class="headerlink" title="至于整合InterProScan的结果可以使用下面的脚本，当然也可以在java -cp :ext/: es.blast2go.prog.B2GAnnotPipe -ips 来实现，或者将b2g4pipe的输出文件加载到 Blast2GO Basic 中进行后续可视化分析。"></a>至于整合InterProScan的结果可以使用下面的脚本，当然也可以在java -cp <em>:ext/</em>: es.blast2go.prog.B2GAnnotPipe -ips 来实现，或者将b2g4pipe的输出文件加载到 Blast2GO Basic 中进行后续可视化分析。</h1><p>merge_interpro_to_go.pl b2g4pipe/go.annot ../interpro/interpro.tsv &gt; go.annot</p>
<h1 id="Anyone-use-diamond-output-xml-file-for-b2gpipe-·-Issue-79-·-bbuchfink-diamond-·-GitHub"><a href="#Anyone-use-diamond-output-xml-file-for-b2gpipe-·-Issue-79-·-bbuchfink-diamond-·-GitHub" class="headerlink" title="Anyone use diamond output xml file  for b2gpipe? · Issue #79 · bbuchfink/diamond · GitHub"></a>Anyone use diamond output xml file  for b2gpipe? · Issue #79 · bbuchfink/diamond · GitHub</h1><h1 id="Problem-in-uploading-the-Diamond-blastx-results-into-Blast2GO-·-Issue-159-·-bbuchfink-diamond-·-GitHub"><a href="#Problem-in-uploading-the-Diamond-blastx-results-into-Blast2GO-·-Issue-159-·-bbuchfink-diamond-·-GitHub" class="headerlink" title="Problem in uploading the Diamond blastx results into Blast2GO · Issue #159 · bbuchfink/diamond · GitHub"></a>Problem in uploading the Diamond blastx results into Blast2GO · Issue #159 · bbuchfink/diamond · GitHub</h1><h1 id="diamond-to-blast2go-erro-·-Issue-165-·-bbuchfink-diamond-·-GitHub"><a href="#diamond-to-blast2go-erro-·-Issue-165-·-bbuchfink-diamond-·-GitHub" class="headerlink" title="diamond to blast2go erro  · Issue #165 · bbuchfink/diamond · GitHub"></a>diamond to blast2go erro  · Issue #165 · bbuchfink/diamond · GitHub</h1><p>2、InterProScan</p>
<p>对于 InterProScan 可以本地化，也可以用 interProScan5.pl 将序列发送到官方服务器进行注释，资源允许，还是推荐本地化，网络版感觉还是有点慢～一个本地化教程:</p>
<p>一个用interproscan做基因注释的简易教程</p>
<p>3、eggNOG-Mapper （包括网页版和本地化）</p>
<p>（1）关于这个注释方法网页版的推荐自己去摸索吧（就是和NCBI提交类似）=&gt; <a href="http://eggnogdb.embl.de/#/app/emapper" target="_blank" rel="noopener">http://eggnogdb.embl.de/#/app/emapper</a></p>
<p>（2）本地化推荐两个教程吧：</p>
<p>序列功能注释神器：eggNOG-mapper，KEGG/COG/KOG/GO/BiGG 一网打尽 ： 序列功能注释神器：eggNOG-mapper，KEGG/COG/KOG/GO/BiGG 一网打尽 « Biostack.org</p>
<p>应该是最好的eggnog-mapper功能注释教程：应该是最好的eggnog-mapper功能注释教程 - 简书（生信媛）<br>关于eggNOG-Mapper 和 BLAST2GO的小心得：</p>
<p>这些方法我都用过，eggNOG-Mapper 是很多人推荐的，我当然也喜欢网页版的，不耗资源，数度快。相比其他的都快。eggNOG-Mapper自己的文章评测认为其准确性高于BLAST2GO，原理上我也这么认为。但有可能是我物种的特异性还是eggNOG-Mapper目前数据库还不够大，自己注释出的蛋白只有6000左右（25000），而BLAST2GO注释有12000多，所以两者目前我觉得都有优点及不足，关键还是看自己的选择。</p>
<p>前方高能:</p>
<p>对于准确性的个人思考，基于比对，也就是基因相似性的注释方法(blastp,diamond)可能会取决于数据库的大小，因为在omicsbox中有涉及到相似度cut这一步，如果数据库不够大，那么相似度有的可能只为50(在cut值之上)，所以只使用Swiss-Prot等小数据库注释时可能需要提高cut值;当然也会听到直系同源和旁系同源的说法，所以在新版Blast2go中增加了filter GO with Taxid (可能是看到eggnog的文章慌了)这一新模块，减少旁系同源的注释结果。那么如果要在老版的Blast2go中运用这个模块，或许可以换个思路，在比对时，通过Taxid来过滤比对结果。</p>
<p>eggnog的注释，存在两种模式diamond和hummer，前者官方推荐为存在接近物种，也就是在直系同源注释中有更好的表现，hummer则反之. 同样的，只要是比对，就会有cut相似度这一说，数据库的大小及深度(涉及的物种)，就会影响结果。其一个特点在于直系同源注释。这点Blast2go中出了filter with Taxid 来补充。</p>
<p>居于以上的讨论，我个人觉得目前还是blast+interpro的准确性会更高一些，当然我觉得eggnog如果持续开发，以及整合diamond和hummer的结果将会成为新的宠儿。</p>
<p>小提示：</p>
<p>首先，第一次写东西，难免会有粗心的地方，望同行能批评指正；对于行文格式，以后会努力。其次，我个人不太会写程序，所以喜欢收集别人的程序，可以在 Zhanmengtao (zhanmengtao) · GitHub 找到一些可能对你来说感兴趣的程序，包括上面提到的一些。</p>
<p>参考文献</p>
<p><a href="http://blog.sina.com.cn/s/blog_16152d7d70102xnc3.html" target="_blank" rel="noopener">http://blog.sina.com.cn/s/blog_16152d7d70102xnc3.html</a></p>
<p><a href="https://github.com/bbuchfink/diamond" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond</a></p>
<p><a href="https://mp.weixin.qq.com/s?__biz=MzI1MjU5MjMzNA==&amp;mid=2247486358&amp;idx=1&amp;sn=4b095e9a200a419079d947d930c38abd&amp;chksm=e9e02237de97ab21f4227ee76eca0aa94433fe6cc374e030093eb3acc2094a546110763bb411&amp;mpshare=1&amp;scene=23&amp;srcid=0101kMbzD2el0RPASBiC9pFi#rd" target="_blank" rel="noopener">https://mp.weixin.qq.com/s?__biz=MzI1MjU5MjMzNA==&amp;mid=2247486358&amp;idx=1&amp;sn=4b095e9a200a419079d947d930c38abd&amp;chksm=e9e02237de97ab21f4227ee76eca0aa94433fe6cc374e030093eb3acc2094a546110763bb411&amp;mpshare=1&amp;scene=23&amp;srcid=0101kMbzD2el0RPASBiC9pFi#rd</a></p>
<p><a href="http://www.jinciwei.cn/a342744.html" target="_blank" rel="noopener">http://www.jinciwei.cn/a342744.html</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/79" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/79</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/159" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/159</a></p>
<p><a href="https://github.com/bbuchfink/diamond/issues/165" target="_blank" rel="noopener">https://github.com/bbuchfink/diamond/issues/165</a></p>
<p><a href="http://www.biostack.org/?p=698" target="_blank" rel="noopener">http://www.biostack.org/?p=698</a></p>
<p><a href="https://www.jianshu.com/p/e646c0fa6443" target="_blank" rel="noopener">https://www.jianshu.com/p/e646c0fa6443</a></p>
<p>NGS 生物信息学分析 V6.0 陈连福 郑越</p>
<p><a href="https://www.jianshu.com/p/6296385adf21-" target="_blank" rel="noopener">https://www.jianshu.com/p/6296385adf21-</a></p>

      
       <hr><span style="font-style: italic;color: gray;"> 转载请注明来源，欢迎对文章中的引用来源进行考证，欢迎指出任何有错误或不够清晰的表达。可以在下面评论区评论，也可以邮件至 zhanmengtao123@163.com </span>
    </div>
</article>


<p>
    <a href="javascript:void(0)" class="dashang" onclick="dashangToggle()">赏</a>
</p>


<div class="article_copyright">
    <p><span class="copy-title">文章标题:</span></p>
    
    <p><span class="copy-title">本文作者:</span><a href="javascript:void(0)" title="詹梦涛">詹梦涛</a></p>
    <p><span class="copy-title">发布时间:</span>2019-09-16, 17:28:04</p>
    <p><span class="copy-title">最后更新:</span>2019-09-16, 17:30:05</p>
    <span class="copy-title">原始链接:</span><a class="post-url" href="/2019/09/16/De-novo-基因组GO-注释小窗口/" title="">http://yoursite.com/2019/09/16/De-novo-基因组GO-注释小窗口/</a>
    <p>
        <span class="copy-title">版权声明:</span><i class="fa fa-creative-commons"></i> <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/" title="CC BY-NC-SA 4.0 International" target = "_blank">"署名-非商用-相同方式共享 4.0"</a> 转载请保留原文链接及作者。
    </p>
</div>





    




    </div>
    <div class="copyright">
        <p class="footer-entry">©2016-2019 Yelog</p>
<p class="footer-entry">Built with <a href="https://hexo.io/" target="_blank">Hexo</a> and <a href="https://github.com/yelog/hexo-theme-3-hexo" target="_blank">3-hexo</a> theme</p>

    </div>
    <div class="full-toc">
        <button class="full"><span class="min "></span></button>
<button class="post-toc-menu"><span class="post-toc-menu-icons"></span></button>
<div class="post-toc"><span class="post-toc-title">目录</span>
    <div class="post-toc-content">

    </div>
</div>
<a class="" id="rocket" href="javascript:void(0)"></a>
    </div>
</div>
<div class="acParent"></div>

<div class="hide_box" onclick="dashangToggle()"></div>
<div class="shang_box">
    <a class="shang_close" href="javascript:void(0)" onclick="dashangToggle()">×</a>
    <div class="shang_tit">
        <p>喜欢就点赞,疼爱就打赏</p>
    </div>
    <div class="shang_payimg">
        <div class="pay_img">
            <img src="/img/alipay.jpg" class="alipay" title="扫码支持">
            <img src="/img/weixin.jpg" class="weixin" title="扫码支持">
        </div>
    </div>
    <div class="shang_payselect">
        <span><label><input type="radio" name="pay" checked value="alipay">支付宝</label></span><span><label><input type="radio" name="pay" value="weixin">微信</label></span>
    </div>
</div>


</body>
<script src="/js/jquery.pjax.js?v=1.0.1" ></script>

<script src="/js/script.js?v=1.0.1" ></script>
<script>
    var img_resize = 'default';
    /*作者、标签的自动补全*/
    $(function () {
        $('.search').AutoComplete({
            'data': [],
            'itemHeight': 20,
            'width': 418
        }).AutoComplete('show');
    })
    function initArticle() {
        /*渲染对应的表格样式*/
        
            $(".post .pjax table").addClass("green_title");
        

        /*渲染打赏样式*/
        
        $("input[name=pay]").on("click", function () {
            if($("input[name=pay]:checked").val()=="weixin"){
                $(".shang_box .shang_payimg .pay_img").addClass("weixin_img");
            } else {
                $(".shang_box .shang_payimg .pay_img").removeClass("weixin_img");
            }
        })
        

        /*高亮代码块行号*/
        
        $('pre code').each(function(){
            var lines = $(this).text().split('\n').length - 1, widther='';
            if (lines>99) {
                widther = 'widther'
            }
            var $numbering = $('<ul/>').addClass('pre-numbering ' + widther).attr("unselectable","on");
            $(this).addClass('has-numbering ' + widther)
                    .parent()
                    .append($numbering);
            for(var i=1;i<=lines;i++){
                $numbering.append($('<li/>').text(i));
            }
        });
        

        /*访问数量*/
        
        $.getScript("//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js");
        

        /*代码高亮，行号对齐*/
        $('.pre-numbering').css('line-height',$('.has-numbering').css('line-height'));

        
    }

    /*打赏页面隐藏与展示*/
    
    function dashangToggle() {
        $(".shang_box").fadeToggle();
        $(".hide_box").fadeToggle();
    }
    

</script>

<!--加入行号的高亮代码块样式-->

<style>
    pre{
        position: relative;
        margin-bottom: 24px;
        border-radius: 10px;
        border: 1px solid #e2dede;
        background: #FFF;
        overflow: hidden;
    }
    code.has-numbering{
        margin-left: 30px;
    }
    code.has-numbering.widther{
        margin-left: 35px;
    }
    .pre-numbering{
        margin: 0px;
        position: absolute;
        top: 0;
        left: 0;
        width: 20px;
        padding: 0.5em 3px 0.7em 5px;
        border-right: 1px solid #C3CCD0;
        text-align: right;
        color: #AAA;
        background-color: #fafafa;
    }
    .pre-numbering.widther {
        width: 35px;
    }
</style>

<!--自定义样式设置-->
<style>
    
    
    .nav {
        width: 542px;
    }
    .nav.fullscreen {
        margin-left: -542px;
    }
    .nav-left {
        width: 120px;
    }
    
    
    @media screen and (max-width: 1468px) {
        .nav {
            width: 492px;
        }
        .nav.fullscreen {
            margin-left: -492px;
        }
        .nav-left {
            width: 100px;
        }
    }
    
    
    @media screen and (max-width: 1024px) {
        .nav {
            width: 492px;
            margin-left: -492px
        }
        .nav.fullscreen {
            margin-left: 0;
        }
        .nav .hide-list.fullscreen {
            left: 492px
        }
    }
    
    @media screen and (max-width: 426px) {
        .nav {
            width: 100%;
        }
        .nav-left {
            width: 100%;
        }
    }
    
    
    .nav-right .title-list nav a .post-title, .nav-right .title-list #local-search-result a .post-title {
        color: #383636;
    }
    
    
    .nav-right .title-list nav a .post-date, .nav-right .title-list #local-search-result a .post-date {
        color: #5e5e5f;
    }
    
    
    .nav-right nav a.hover, #local-search-result a.hover{
        background-color: #e2e0e0;
    }
    
    

    /*列表样式*/
    
    .post .pjax article .article-entry>ol, .post .pjax article .article-entry>ul, .post .pjax article>ol, .post .pjax article>ul{
        border: #e2dede solid 1px;
        border-radius: 10px;
        padding: 10px 32px 10px 56px;
    }
    .post .pjax article .article-entry li>ol, .post .pjax article .article-entry li>ul,.post .pjax article li>ol, .post .pjax article li>ul{
        padding-top: 5px;
        padding-bottom: 5px;
    }
    .post .pjax article .article-entry>ol>li, .post .pjax article .article-entry>ul>li,.post .pjax article>ol>li, .post .pjax article>ul>li{
        margin-bottom: auto;
        margin-left: auto;
    }
    .post .pjax article .article-entry li>ol>li, .post .pjax article .article-entry li>ul>li,.post .pjax article li>ol>li, .post .pjax article li>ul>li{
        margin-bottom: auto;
        margin-left: auto;
    }
    

    /* 背景图样式 */
    
    


    /*引用块样式*/
    

    /*文章列表背景图*/
    
    .nav-right:before {
        content: ' ';
        display: block;
        position: absolute;
        left: 0;
        top: 0;
        width: 100%;
        height: 100%;
        opacity: 0.3;
        background: url("https://i.loli.net/2019/07/22/5d3521411f3f169375.png");
        background-repeat: no-repeat;
        background-position: 50% 0;
        -ms-background-size: cover;
        -o-background-size: cover;
        -moz-background-size: cover;
        -webkit-background-size: cover;
        background-size: cover;
    }
    

    
</style>







</html>
