<!DOCTYPE html>
<html lang="zh">
<head><meta name="generator" content="Hexo 3.9.0">
    <meta charset="utf-8">
<title>一个高杂合真菌基因组组装脚本 - 詹梦涛的生信博客</title>
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">



    <meta name="description" content="categories:    - assemble 前言及背景什么是基因组de novo测序？其是对某一物种进行高通量测序，利用高性能计算平台和生物信息学方法，在不依赖于参考基因组的情况下进行组装，从而绘制该物种的全基因组序列图谱。 针对基因组的特性，基因组常被分为两类：普通（简单）基因组和复杂基因组。简单基因组指单倍体，纯合二倍体或者杂合度&amp;lt;0.5%，且重复序列含量&amp;lt;50%，GC含量">
<meta property="og:type" content="article">
<meta property="og:title" content="一个高杂合真菌基因组组装脚本">
<meta property="og:url" content="http://www.zhanmengtao.top/2019/10/30/复杂基因组组装/index.html">
<meta property="og:site_name" content="詹梦涛的生信博客">
<meta property="og:description" content="categories:    - assemble 前言及背景什么是基因组de novo测序？其是对某一物种进行高通量测序，利用高性能计算平台和生物信息学方法，在不依赖于参考基因组的情况下进行组装，从而绘制该物种的全基因组序列图谱。 针对基因组的特性，基因组常被分为两类：普通（简单）基因组和复杂基因组。简单基因组指单倍体，纯合二倍体或者杂合度&amp;lt;0.5%，且重复序列含量&amp;lt;50%，GC含量">
<meta property="og:locale" content="zh">
<meta property="og:image" content="http://www.zhanmengtao.top/images/og_image.png">
<meta property="og:updated_time" content="2019-10-30T15:45:30.156Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="一个高杂合真菌基因组组装脚本">
<meta name="twitter:description" content="categories:    - assemble 前言及背景什么是基因组de novo测序？其是对某一物种进行高通量测序，利用高性能计算平台和生物信息学方法，在不依赖于参考基因组的情况下进行组装，从而绘制该物种的全基因组序列图谱。 针对基因组的特性，基因组常被分为两类：普通（简单）基因组和复杂基因组。简单基因组指单倍体，纯合二倍体或者杂合度&amp;lt;0.5%，且重复序列含量&amp;lt;50%，GC含量">
<meta name="twitter:image" content="http://www.zhanmengtao.top/images/og_image.png">







<link rel="icon" href="/images/favicon.svg">


<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.7.2/css/bulma.css">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.4.1/css/all.css">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:400,600|Source+Code+Pro">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@9.12.0/styles/atom-one-light.css">


    
    
    
    <style>body>.footer,body>.navbar,body>.section{opacity:0}</style>
    

    
    
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/css/lightgallery.min.css">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/css/justifiedGallery.min.css">
    

    
    

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/outdatedbrowser@1.1.5/outdatedbrowser/outdatedbrowser.min.css">


    
    
    
    

<link rel="stylesheet" href="/css/back-to-top.css">


    
    

    
    
    
    

    
    
<link rel="stylesheet" href="/css/progressbar.css">
<script src="https://cdn.jsdelivr.net/npm/pace-js@1.0.2/pace.min.js"></script>

    
    
    


<link rel="stylesheet" href="/css/style.css">
</head>
<body class="is-3-column">
    <nav class="navbar navbar-main">
    <div class="container">
        <div class="navbar-brand is-flex-center">
            <a class="navbar-item navbar-logo" href="/">
            
                <img src="/images/og_image.png" alt="一个高杂合真菌基因组组装脚本" height="28">
            
            </a>
        </div>
        <div class="navbar-menu">
            
            <div class="navbar-start">
                
                <a class="navbar-item" href="/">Home</a>
                
                <a class="navbar-item" href="/archives">Archives</a>
                
                <a class="navbar-item" href="/categories">Categories</a>
                
                <a class="navbar-item" href="/tags">Tags</a>
                
                <a class="navbar-item" href="/about">About</a>
                
            </div>
            
            <div class="navbar-end">
                
                    
                    
                    <a class="navbar-item" target="_blank" title="Download on GitHub" href="https://github.com/ppoffice/hexo-theme-icarus">
                        
                        <i class="fab fa-github"></i>
                        
                    </a>
                    
                
                
                
                <a class="navbar-item search" title="Search" href="javascript:;">
                    <i class="fas fa-search"></i>
                </a>
                
            </div>
        </div>
    </div>
</nav>
    
    <section class="section">
        <div class="container">
            <div class="columns">
                <div class="column is-8-tablet is-8-desktop is-6-widescreen has-order-2 column-main"><div class="card">
    
    <div class="card-content article ">
        
        <div class="level article-meta is-size-7 is-uppercase is-mobile is-overflow-x-auto">
            <div class="level-left">
                <time class="level-item has-text-grey" datetime="2019-10-30T15:13:32.836Z">2019-10-30</time>
                
                
                <span class="level-item has-text-grey">
                    
                    
                    16 minutes read (About 2349 words)
                </span>
                
                
            </div>
        </div>
        
        <h1 class="title is-size-3 is-size-4-mobile has-text-weight-normal">
            
                一个高杂合真菌基因组组装脚本
            
        </h1>
        <div class="content">
            <p>categories:<br>    - assemble</p>
<h3 id="前言及背景"><a href="#前言及背景" class="headerlink" title="前言及背景"></a>前言及背景</h3><p>什么是基因组de novo测序？其是对某一物种进行高通量测序，利用高性能计算平台和生物信息学方法，在不依赖于参考基因组的情况下进行组装，从而绘制该物种的全基因组序列图谱。</p>
<p>针对基因组的特性，基因组常被分为两类：普通（简单）基因组和复杂基因组。简单基因组指单倍体，纯合二倍体或者杂合度&lt;0.5%，且重复序列含量&lt;50%，GC含量为35%到65%之间的二倍体。复杂基因组则指杂合率＞0.5%，重复序列含量＞50%，GC含量处于异常的范围（GC含量＜35%或者GC含量＞＝65%的二倍体，多倍体。诺禾致源对二倍体复杂基因组进一步细分为微杂合基因组（0.5%＜杂合率＜＝0.8%、高杂合基因组（杂合率＞0.8%）以及高重复基因组（重复序列比例&gt;50%）。复杂基因组的组装一直以来都是一个让科研工作者为之头疼的问题。科研工作者也为解决这个问题一直努力着。随着三代测序平台的更迭，测序subreads的长度不断的增长，以及光学图谱技术的出现（Hic,10xGenomic等）。重复序列导致的组装困难逐步被缓解。但高杂合这一特性任一直困扰着科研工作者。</p>
<p>为了解决杂合组装，各式各样的方案不断被提出。总体思路有下：1.设计实验方案获取单倍体（例如种间杂种；案例：<a href="https://www.nature.com/articles/s41438-019-0139-1" target="_blank" rel="noopener">Sequencing a <em>Juglans regia</em> × <em>J. microcarpa</em> hybrid yields high-quality genome assemblies of parental species</a>）；2.设计适合于杂合基因组组装的软件，优化组装的算法；3.组装之后，再用去杂合软件去除杂合序列。</p>
<p>第一个思路很清晰，获取单倍体再测序，就解决了高杂合这个难点，可有时单倍体的获取真不是一般的难，局限性很大。</p>
<p>基于第二个思路，已有很多软件开发。有NOVOheter, Plantanus, MSR-CA，Plantanus-allee（Plantanus的升级版，支持Hic，10xGenomics等数据）等。此外还有一些软件有支持高杂合组装的模块，如Canu，SPAdes, Falon等。但实测的经验来看，效果都不是很好。</p>
<p>第三种思路的核心就是，居于相似性cut，目前接触过的的有Redundans，Haplomerger2，Purge_haplogs。Purge_haplogs除了考虑相似性，还有一大特性，就是通过分析比对read的覆盖度决定谁去谁留。此外，Haplomerger2和Purge_haplogs还支持重复序列部分的屏蔽。</p>
<p>今年我接手到了一个基因组大小为86M,杂合率约2.4%，重复序列率约为20%左右的真菌。测序策略为pacbio seq I + PE 150 。 本来打算再做一个Hic，但咨询一些测序公司之后，均表示真菌没有太多成功经验，且投入与产出可能不对等。故打消测Hic的念头。</p>
<p>基于已有的数据，我先采取了canu 单倍型组装、canu(多倍体模式，”batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50”组装，MECAT2，MaSuRCA、Falon、flye、wtdbg2 等组装软件进行了组装。接着使用 Purge_haplogs 、Redundans和Haplomerger2进行去杂合，然后再使用 FinisherSC进行基因组升级，最后使用nextpolish 进行 polish。经过测试，canu，MECAT2（经过nextpolish 进行 polish之后）， MaSuRCA的三个软件的表现相对较好。 去杂软件Purge_haplogs的适用性优于Redundans和Haplomerger2，去杂前后BUSCO评估基本不变。由于涉及到文章。故暂时只能提供最优的脚本。</p>
<h3 id="组装案例"><a href="#组装案例" class="headerlink" title="组装案例"></a>组装案例</h3><figure class="highlight sh hljs"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br></pre></td><td class="code"><pre><span class="line"><span class="hljs-comment"># 测序数据 Bam to Fastq or Fasta</span></span><br><span class="line">samtools fastq -0 012m.subreads.fq -@ 32 subreads.bam</span><br><span class="line"><span class="hljs-comment"># Assessment of genome size and heterozygosity(raw PE reads)</span></span><br><span class="line">mkdir genomescope</span><br><span class="line"><span class="hljs-built_in">cd</span> genomescope</span><br><span class="line">ln -s ../012m_L1_?.fq ./</span><br><span class="line">jellyfish count -C -m 21 -s 1000000000 -t 32 *.fq -o reads.jf</span><br><span class="line">jellyfish histo -t 32 reads.jf &gt; reads.histo</span><br><span class="line">Rscript /opt/biosoft/genomescope/genomescope.R reads.histo 21 150 output</span><br><span class="line"><span class="hljs-comment"># 二代数据质控</span></span><br><span class="line">mkdir Trimmomatic</span><br><span class="line"><span class="hljs-built_in">cd</span> Trimmomatic</span><br><span class="line">java -jar /opt/biosoft/Trimmomatic-0.36/trimmomatic-0.36.jar PE -threads 32 -phred33 ../012m_L1_1.fq ../012m_L1_2.fq 012m_Trimmomatic.1.fq 012m_Trimmomatic.unpaired.1.fq 012m_Trimmomatic.2.fq 012m_Trimmomatic.unpaired.2.fq \</span><br><span class="line">ILLUMINACLIP:/opt/biosoft/Trimmomatic-0.36/adapters/TruSeq3-PE-2.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 MINLEN:75 TOPHRED33</span><br><span class="line"><span class="hljs-built_in">cd</span> ../</span><br><span class="line"><span class="hljs-comment"># correct Illumina reads | Assessment of genome size and heterozygosity</span></span><br><span class="line">mkdir Finderrors</span><br><span class="line"><span class="hljs-built_in">source</span> ~/.bash.pacbio</span><br><span class="line">ErrorCorrectReads.pl PHRED_ENCODING=33 READS_OUT=012m FILL_FRAGMENTS=0 KEEP_KMER_SPECTRA=1 PAIRED_READS_A_IN=../fastuniq/012m.fastuniq.1.fastq PAIRED_READS_B_IN=../fastuniq/012m.fastuniq.1.fastq PLOIDY=2 PAIRED_SEP=251 PAIRED_STDEV=48</span><br><span class="line"><span class="hljs-comment"># ErrorCorrectReads.pl也可以对基因组特性进行评估，但针对我的菌株来说，同已发布的邻近菌株比较，GenomeScope的结果要准确一些。</span></span><br><span class="line"></span><br><span class="line"><span class="hljs-comment"># kmer plot</span></span><br><span class="line"><span class="hljs-built_in">cd</span> 012m.fastq.kspec</span><br><span class="line">perl -p -i -e <span class="hljs-string">'s/\@fns = \(\"frag_reads_filt.25mer.kspec\", \"frag_reads_edit.24mer.kspec\", \"frag_reads_corr.25mer.kspec\"\);\n//'</span> /opt/biosoft/ALLPATHS-LG/bin/KmerSpectrumPlot.pl</span><br><span class="line">KmerSpectrumPlot.pl SPECTRA=1 FREQ_MAX=255</span><br><span class="line">perl -p -i -e <span class="hljs-string">'s/\@fns = \(\"frag_reads_filt.25mer.kspec\", \"frag_reads_edit.24mer.kspec\", \"frag_reads_corr.25mer.kspec\"\);\n//'</span> /opt/biosoft/ALLPATHS-LG/bin/KmerSpectrumPlot.pl</span><br><span class="line">convert kmer_spectrum.cumulative_frac.log.lin.eps kmer_spectrum.cumulative_frac.log.lin.png</span><br><span class="line">convert kmer_spectrum.distinct.lin.lin.eps kmer_spectrum.distinct.lin.lin.png</span><br><span class="line">convert kmer_spectrum.distinct.log.log.eps kmer_spectrum.distinct.log.log.png</span><br><span class="line"><span class="hljs-built_in">cd</span> ../</span><br><span class="line"><span class="hljs-comment"># Using LoRDEC to modify PacBio Reads</span></span><br><span class="line">mkdir LoRDEC</span><br><span class="line"><span class="hljs-built_in">cd</span> LoRDEC</span><br><span class="line">lordec-correct -2 ../Finderrors/012m.paired.A.fastq ../Finderrors/012m.paired.B.fastq -i ../012m.subreads.fq -k 19 -o pacbio.LoRDEC.corrected.fasta -s 3 -T 32 &amp;&gt; lordec-correct.log</span><br><span class="line">seqkit seq -u pacbio.LoRDEC.corrected.fasta &gt; pacbio.corrected.fasta</span><br><span class="line"><span class="hljs-built_in">cd</span> ../</span><br><span class="line"><span class="hljs-comment">############# Genome assembly ###############</span></span><br><span class="line">mkdir MaSuRCA</span><br><span class="line"><span class="hljs-built_in">cd</span> MaSuRCA</span><br><span class="line"><span class="hljs-built_in">echo</span> <span class="hljs-string">'# example configuration file </span></span><br><span class="line"><span class="hljs-string">DATA</span></span><br><span class="line"><span class="hljs-string"># Illumina paired end reads supplied as &lt;two-character prefix&gt; &lt;fragment mean&gt; &lt;fragment stdev&gt; &lt;forward_reads&gt; &lt;reverse_reads&gt;</span></span><br><span class="line"><span class="hljs-string"># if single-end, do not specify &lt;reverse_reads&gt;</span></span><br><span class="line"><span class="hljs-string"># MUST HAVE Illumina paired end reads to use MaSuRCA</span></span><br><span class="line"><span class="hljs-string">PE= pe 251 48 /home/bioinfo/data/012m/012m_L1_1.fq  /home/bioinfo/data/012m/012m_L1_2.fq</span></span><br><span class="line"><span class="hljs-string"># Illumina mate pair reads supplied as &lt;two-character prefix&gt; &lt;fragment mean&gt; &lt;fragment stdev&gt; &lt;</span></span><br><span class="line"><span class="hljs-string">forward_reads&gt; &lt;reverse_reads&gt;</span></span><br><span class="line"><span class="hljs-string"># JUMP= sh 3600 200  /FULL_PATH/short_1.fastq  /FULL_PATH/short_2.fastq</span></span><br><span class="line"><span class="hljs-string"># pacbio OR nanopore reads must be in a single fasta or fastq file with absolute path, can be gzipped</span></span><br><span class="line"><span class="hljs-string"># if you have both types of reads supply them both as NANOPORE type</span></span><br><span class="line"><span class="hljs-string">PACBIO=/home/bioinfo/data/012m/LoRDEC/pacbio.corrected.fasta</span></span><br><span class="line"><span class="hljs-string"># NANOPORE=/FULL_PATH/nanopore.fa</span></span><br><span class="line"><span class="hljs-string"># Other reads (Sanger, 454, etc) one frg file, concatenate your frg files into one if you have many</span></span><br><span class="line"><span class="hljs-string"># OTHER=/FULL_PATH/file.frg</span></span><br><span class="line"><span class="hljs-string"># synteny-assisted assembly, concatenate all reference genomes into one reference.fa; works for Illumina-only data</span></span><br><span class="line"><span class="hljs-string"># REFERENCE=/FULL_PATH/nanopore.fa</span></span><br><span class="line"><span class="hljs-string">END</span></span><br><span class="line"><span class="hljs-string">PARAMETERS</span></span><br><span class="line"><span class="hljs-string"># PLEASE READ all comments to essential parameters below, and set the parameters according to your project</span></span><br><span class="line"><span class="hljs-string"># set this to 1 if your Illumina jumping library reads are shorter than 100bp</span></span><br><span class="line"><span class="hljs-string">EXTEND_JUMP_READS=0</span></span><br><span class="line"><span class="hljs-string"># this is k-mer size for deBruijn graph values between 25 and 127 are supported, auto will compute the optimal size based on the read data and GC content</span></span><br><span class="line"><span class="hljs-string">GRAPH_KMER_SIZE = auto</span></span><br><span class="line"><span class="hljs-string"># set this to 1 for all Illumina-only assemblies</span></span><br><span class="line"><span class="hljs-string"># set this to 0 if you have more than 15x coverage by long reads (Pacbio or Nanopore) or any other long reads/mate pairs (Illumina MP, Sanger, 454, etc)</span></span><br><span class="line"><span class="hljs-string">USE_LINKING_MATES = 0</span></span><br><span class="line"><span class="hljs-string"># specifies whether to run the assembly on the grid</span></span><br><span class="line"><span class="hljs-string">USE_GRID=0</span></span><br><span class="line"><span class="hljs-string"># specifies grid engine to use SGE or SLURM</span></span><br><span class="line"><span class="hljs-string">GRID_ENGINE=SGE</span></span><br><span class="line"><span class="hljs-string"># specifies queue (for SGE) or partition (for SLURM) to use when running on the grid MANDATORY</span></span><br><span class="line"><span class="hljs-string">GRID_QUEUE=all.q</span></span><br><span class="line"><span class="hljs-string"># batch size in the amount of long read sequence for each batch on the grid</span></span><br><span class="line"><span class="hljs-string">GRID_BATCH_SIZE=500000000</span></span><br><span class="line"><span class="hljs-string"># use at most this much coverage by the longest Pacbio or Nanopore reads, discard the rest of the reads</span></span><br><span class="line"><span class="hljs-string"># can increase this to 30 or 35 if your reads are short (N50&lt;7000bp)</span></span><br><span class="line"><span class="hljs-string">LHE_COVERAGE=25</span></span><br><span class="line"><span class="hljs-string"># set to 0 (default) to do two passes of mega-reads for slower, but higher quality assembly, otherwise set to 1</span></span><br><span class="line"><span class="hljs-string">MEGA_READS_ONE_PASS=0</span></span><br><span class="line"><span class="hljs-string"># this parameter is useful if you have too many Illumina jumping library mates. Typically set it to 60 for bacteria and 300 for the other organisms </span></span><br><span class="line"><span class="hljs-string"># LIMIT_JUMP_COVERAGE = 300</span></span><br><span class="line"><span class="hljs-string"># these are the additional parameters to Celera Assembler.  do not worry about performance, number or processors or batch sizes -- these are computed automatically. </span></span><br><span class="line"><span class="hljs-string"># CABOG ASSEMBLY ONLY: set cgwErrorRate=0.25 for bacteria and 0.1&lt;=cgwErrorRate&lt;=0.15 for other organisms.</span></span><br><span class="line"><span class="hljs-string">CA_PARAMETERS =  cgwErrorRate=0.15</span></span><br><span class="line"><span class="hljs-string"># CABOG ASSEMBLY ONLY: whether to attempt to close gaps in scaffolds with Illumina  or long read data</span></span><br><span class="line"><span class="hljs-string">CLOSE_GAPS=1</span></span><br><span class="line"><span class="hljs-string"># auto-detected number of cpus to use, set this to the number of CPUs/threads per node you will be using</span></span><br><span class="line"><span class="hljs-string">NUM_THREADS = 32</span></span><br><span class="line"><span class="hljs-string"># this is mandatory jellyfish hash size -- a safe value is estimated_genome_size*20</span></span><br><span class="line"><span class="hljs-string">JF_SIZE = 1740000000</span></span><br><span class="line"><span class="hljs-string"># ILLUMINA ONLY. Set this to 1 to use SOAPdenovo contigging/scaffolding module.  Assembly will be worse but will run faster. Useful for very large (&gt;=8Gbp) genomes from Illumina-only data</span></span><br><span class="line"><span class="hljs-string">SOAP_ASSEMBLY=0</span></span><br><span class="line"><span class="hljs-string">#Hybrid Illumina paired end + Nanopore/PacBio assembly ONLY.  Set this to 1 to use Flye assembler for final assembly of corrected mega-reads.  A lot faster than CABOG, at the expense of some contiguity. Works well even when MEGA_READS_ONE_PASS is set to 1.  DO NOT use if you have less than 15x coverage by long reads.</span></span><br><span class="line"><span class="hljs-string">FLYE_ASSEMBLY=0</span></span><br><span class="line"><span class="hljs-string">END '</span> &gt; config.txt</span><br><span class="line">/opt/biosoft/MaSuRCA-3.3.4/bin/masurca config.txt</span><br><span class="line">./assemble.sh</span><br><span class="line"></span><br><span class="line">mkdir purge_haplogs</span><br><span class="line"><span class="hljs-built_in">cd</span> purge_haplogs</span><br><span class="line">minimap2 -t 32 -ax map-pb ../final.genome.scf.fasta\</span><br><span class="line">/home/bioinfo/data/012m/canu/012m.correctedReads.fasta.gz | samtools view -hF 256 - | samtools sort -@ 32 -m 2G -o aligned.bam</span><br><span class="line">purge_haplotigs  hist  -b aligned.bam  -g ../final.genome.scf.fasta -t 20</span><br><span class="line">purge_haplotigs contigcov -i aligned.bam.gencov -o coverage_stats.csv -l 18 -m 76 -h 134</span><br><span class="line">purge_haplotigs purge -g ../final.genome.scf.fasta -c coverage_stats.csv -b aligned.bam -t 4 -a 50</span><br><span class="line"></span><br><span class="line">mkdir finisherSC</span><br><span class="line"><span class="hljs-built_in">cd</span> finisherSC</span><br><span class="line">ln -s ../curated.fasta ./contigs.fasta</span><br><span class="line">ln -s ~/data/012m/canu/012m.correctedReads.fasta ./raw_reads.fasta</span><br><span class="line">python /opt/biosoft/finishingTool/finisherSC.py ./ /opt/biosoft/mummer4/bin/</span><br><span class="line"></span><br><span class="line">mkdir NextPolish</span><br><span class="line"><span class="hljs-built_in">cd</span> NextPolish</span><br><span class="line">ls ~/data/012m/Finderrors/012m.paired.?.fastq &gt; sgs.fofn</span><br><span class="line"><span class="hljs-built_in">echo</span> <span class="hljs-string">'/home/bioinfo/data/012m/canu/012m.correctedReads.fasta'</span> &gt; lgs.fofn</span><br><span class="line"><span class="hljs-built_in">echo</span> <span class="hljs-string">'[General]</span></span><br><span class="line"><span class="hljs-string">job_type = local</span></span><br><span class="line"><span class="hljs-string">job_prefix = nextPolish</span></span><br><span class="line"><span class="hljs-string">task = default</span></span><br><span class="line"><span class="hljs-string">rewrite = yes</span></span><br><span class="line"><span class="hljs-string">rerun = 10</span></span><br><span class="line"><span class="hljs-string">parallel_jobs = 5</span></span><br><span class="line"><span class="hljs-string">multithread_jobs = 6</span></span><br><span class="line"><span class="hljs-string">genome = ../improved3.fasta</span></span><br><span class="line"><span class="hljs-string">genome_size = auto</span></span><br><span class="line"><span class="hljs-string">workdir = ./01_rundir</span></span><br><span class="line"><span class="hljs-string">polish_options = -p &#123;multithread_jobs&#125;</span></span><br><span class="line"><span class="hljs-string"></span></span><br><span class="line"><span class="hljs-string">[sgs_option]</span></span><br><span class="line"><span class="hljs-string">sgs_fofn = ./sgs.fofn</span></span><br><span class="line"><span class="hljs-string">sgs_options = -max_depth 100 -bwa</span></span><br><span class="line"><span class="hljs-string"></span></span><br><span class="line"><span class="hljs-string">[lgs_option]</span></span><br><span class="line"><span class="hljs-string">lgs_fofn = ./lgs.fofn</span></span><br><span class="line"><span class="hljs-string">lgs_options = -min_read_len 10k -max_read_len 150k -max_depth 60</span></span><br><span class="line"><span class="hljs-string">lgs_minimap2_options = -x map-pb</span></span><br><span class="line"><span class="hljs-string"></span></span><br><span class="line"><span class="hljs-string">[polish_options]</span></span><br><span class="line"><span class="hljs-string">-ploidy 2 '</span> &gt; run.cfg</span><br><span class="line"></span><br><span class="line">/opt/biosoft/NextPolish/nextPolish run.cfg</span><br><span class="line"></span><br><span class="line">cat ./01_rundir/01.kmer_count/*polish.ref.sh.work/polish_genome*/genome.nextpolish.part*.fasta &gt; genome.nextpolish.fasta</span><br></pre></td></tr></table></figure>

<h5 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h5><p><a href="https://wenku.baidu.com/view/737ef614f46527d3240ce0f1.html" target="_blank" rel="noopener">动植物基因组de novo常见问题</a><br><a href="http://www.wanfangdata.com.cn/details/detail.do?_type=perio&id=yc201811004" target="_blank" rel="noopener">杂基因组测序技术研究进展</a><br><a href="https://github.com/Nextomics/NextPolish/blob/master/doc/OPTION.md" target="_blank" rel="noopener">NextPolish</a><br><a href="https://www.nature.com/articles/s41438-019-0139-1" target="_blank" rel="noopener">Sequencing a <em>Juglans regia</em> × <em>J. microcarpa</em> hybrid yields high-quality genome assemblies of parental species</a><br><a href="https://www.jianshu.com/p/154584a3556f" target="_blank" rel="noopener">LoRDEC 利用二代数据纠错PacBio 数据</a><br><a href="https://github.com/schatzlab/genomescope" target="_blank" rel="noopener">genomescope</a><br><a href="https://github.com/alekseyzimin/masurca" target="_blank" rel="noopener">MaSuRCA</a><br><a href="https://bitbucket.org/mroachawri/purge_haplotigs/" target="_blank" rel="noopener">purge_haplogs</a></p>

        </div>
        
        
        
    </div>
</div>



<div class="card">
    <div class="card-content">
        <h3 class="menu-label has-text-centered">Like this article? Support the author with</h3>
        <div class="buttons is-centered">
            
                
<a class="button is-info donate">
    <span class="icon is-small">
        <i class="fab fa-alipay"></i>
    </span>
    <span>Alipay</span>
    <div class="qrcode"><img src="/images/alipay.jpg" alt="Alipay"></div>
</a>

                
                
<a class="button is-success donate">
    <span class="icon is-small">
        <i class="fab fa-weixin"></i>
    </span>
    <span>Wechat</span>
    <div class="qrcode"><img src="/images/weixin.jpg" alt="Wechat"></div>
</a>

                
        </div>
    </div>
</div>



<div class="card card-transparent">
    <div class="level post-navigation is-flex-wrap is-mobile">
        
        
        <div class="level-end">
            <a class="level level-item has-link-grey  article-nav-next" href="/2019/09/16/GO-annotation/">
                <span class="level-item">De novo 基因组 GO 注释小窗口</span>
                <i class="level-item fas fa-chevron-right"></i>
            </a>
        </div>
        
    </div>
</div>


</div>
                




<div class="column is-4-tablet is-4-desktop is-3-widescreen  has-order-1 column-left ">
    
        
<div class="card widget">
    <div class="card-content">
        <nav class="level">
            <div class="level-item has-text-centered" style="flex-shrink: 1">
                <div>
                    
                        <img class="image is-128x128 has-mb-6" src="/images/avatar.jpg" alt="詹梦涛">
                    
                    
                    <p class="is-size-4 is-block">
                        詹梦涛
                    </p>
                    
                    
                    <p class="is-size-6 is-block">
                        Happy Boy
                    </p>
                    
                    
                    <p class="is-size-6 is-flex is-flex-center has-text-grey">
                        <i class="fas fa-map-marker-alt has-mr-7"></i>
                        <span>Kunming Yunnan China</span>
                    </p>
                    
                </div>
            </div>
        </nav>
        <nav class="level is-mobile">
            <div class="level-item has-text-centered is-marginless">
                <div>
                    <p class="heading">
                        Entradas
                    </p>
                    <p class="title has-text-weight-normal">
                        3
                    </p>
                </div>
            </div>
            <div class="level-item has-text-centered is-marginless">
                <div>
                    <p class="heading">
                        Categorias
                    </p>
                    <p class="title has-text-weight-normal">
                        0
                    </p>
                </div>
            </div>
            <div class="level-item has-text-centered is-marginless">
                <div>
                    <p class="heading">
                        Etiquetas
                    </p>
                    <p class="title has-text-weight-normal">
                        0
                    </p>
                </div>
            </div>
        </nav>
        <div class="level">
            <a class="level-item button is-link is-rounded" href="https://github.com/Zhanmengtao" target="_blank">
                SEGUIR</a>
        </div>
        
        
        <div class="level is-mobile">
            
            <a class="level-item button is-white is-marginless" target="_blank" title="Github" href="https://github.com/Zhanmengtao">
                
                <i class="fab fa-github"></i>
                
            </a>
            
            <a class="level-item button is-white is-marginless" target="_blank" title="Facebook" href="https://facebook.com">
                
                <i class="fab fa-facebook"></i>
                
            </a>
            
            <a class="level-item button is-white is-marginless" target="_blank" title="Twitter" href="https://twitter.com">
                
                <i class="fab fa-twitter"></i>
                
            </a>
            
            <a class="level-item button is-white is-marginless" target="_blank" title="Dribbble" href="https://dribbble.com">
                
                <i class="fab fa-dribbble"></i>
                
            </a>
            
            <a class="level-item button is-white is-marginless" target="_blank" title="RSS" href="/">
                
                <i class="fas fa-rss"></i>
                
            </a>
            
            <a class="level-item button is-white is-marginless" target="_blank" title="QQ" href="/1406390902">
                
                <i class="fas fa-qq"></i>
                
            </a>
            
        </div>
        
    </div>
</div>
    
        
    
        

<div class="card widget">
    <div class="card-content">
        <div class="menu">
        <h3 class="menu-label">
            Links
        </h3>
        <ul class="menu-list">
        
            <li>
                <a class="level is-mobile" href="http://xuzhougeng.top/" target="_blank">
                    <span class="level-left">
                        <span class="level-item">徐洲更的第二大脑</span>
                    </span>
                    <span class="level-right">
                        <span class="level-item tag">xuzhougeng.top</span>
                    </span>
                </a>
            </li>
        
            <li>
                <a class="level is-mobile" href="https://kaopubear.top/" target="_blank">
                    <span class="level-left">
                        <span class="level-item">思考问题的熊</span>
                    </span>
                    <span class="level-right">
                        <span class="level-item tag">kaopubear.top</span>
                    </span>
                </a>
            </li>
        
            <li>
                <a class="level is-mobile" href="http://guoshipeng.com/" target="_blank">
                    <span class="level-left">
                        <span class="level-item">果子学生信</span>
                    </span>
                    <span class="level-right">
                        <span class="level-item tag">guoshipeng.com</span>
                    </span>
                </a>
            </li>
        
            <li>
                <a class="level is-mobile" href="http://www.bio-info-trainee.com/" target="_blank">
                    <span class="level-left">
                        <span class="level-item">生信菜鸟团</span>
                    </span>
                    <span class="level-right">
                        <span class="level-item tag">www.bio-info-trainee.com</span>
                    </span>
                </a>
            </li>
        
            <li>
                <a class="level is-mobile" href="https://hexo.io" target="_blank">
                    <span class="level-left">
                        <span class="level-item">Hexo</span>
                    </span>
                    <span class="level-right">
                        <span class="level-item tag">hexo.io</span>
                    </span>
                </a>
            </li>
        
            <li>
                <a class="level is-mobile" href="https://github.com/ppoffice" target="_blank">
                    <span class="level-left">
                        <span class="level-item">PPOffice</span>
                    </span>
                    <span class="level-right">
                        <span class="level-item tag">github.com</span>
                    </span>
                </a>
            </li>
        
        </ul>
        </div>
    </div>
</div>


    
        
<div class="card widget">
    <div class="card-content">
        <div class="menu">
            <h3 class="menu-label">
                Categorias
            </h3>
            <ul class="menu-list">
            
            </ul>
        </div>
    </div>
</div>
    
        
    
    
        <div class="column-right-shadow is-hidden-widescreen ">
        
            
<div class="card widget">
    <div class="card-content">
        <h3 class="menu-label">
            Recientes
        </h3>
        
        <article class="media">
            
            <a href="/2019/10/30/复杂基因组组装/" class="media-left">
                <p class="image is-64x64">
                    <img class="thumbnail" src="/images/thumbnail.svg" alt="一个高杂合真菌基因组组装脚本">
                </p>
            </a>
            
            <div class="media-content">
                <div class="content">
                    <div><time class="has-text-grey is-size-7 is-uppercase" datetime="2019-10-30T15:13:32.836Z">2019-10-30</time></div>
                    <a href="/2019/10/30/复杂基因组组装/" class="title has-link-black-ter is-size-6 has-text-weight-normal">一个高杂合真菌基因组组装脚本</a>
                    <p class="is-size-7 is-uppercase">
                        
                    </p>
                </div>
            </div>
        </article>
        
        <article class="media">
            
            <a href="/2019/09/16/GO-annotation/" class="media-left">
                <p class="image is-64x64">
                    <img class="thumbnail" src="/images/thumbnail.svg" alt="De novo 基因组 GO 注释小窗口">
                </p>
            </a>
            
            <div class="media-content">
                <div class="content">
                    <div><time class="has-text-grey is-size-7 is-uppercase" datetime="2019-09-16T12:32:41.938Z">2019-09-16</time></div>
                    <a href="/2019/09/16/GO-annotation/" class="title has-link-black-ter is-size-6 has-text-weight-normal">De novo 基因组 GO 注释小窗口</a>
                    <p class="is-size-7 is-uppercase">
                        
                    </p>
                </div>
            </div>
        </article>
        
        <article class="media">
            
            <a href="/2019/09/16/hello-world/" class="media-left">
                <p class="image is-64x64">
                    <img class="thumbnail" src="/images/thumbnail.svg" alt="Hello World">
                </p>
            </a>
            
            <div class="media-content">
                <div class="content">
                    <div><time class="has-text-grey is-size-7 is-uppercase" datetime="2019-09-16T03:35:12.762Z">2019-09-16</time></div>
                    <a href="/2019/09/16/hello-world/" class="title has-link-black-ter is-size-6 has-text-weight-normal">Hello World</a>
                    <p class="is-size-7 is-uppercase">
                        
                    </p>
                </div>
            </div>
        </article>
        
    </div>
</div>

        
            <div class="card widget">
    <div class="card-content">
        <div class="menu">
        <h3 class="menu-label">
            Archivos
        </h3>
        <ul class="menu-list">
        
        <li>
            <a class="level is-marginless" href="/archives/2019/10/">
                <span class="level-start">
                    <span class="level-item">October 2019</span>
                </span>
                <span class="level-end">
                    <span class="level-item tag">1</span>
                </span>
            </a>
        </li>
        
        <li>
            <a class="level is-marginless" href="/archives/2019/09/">
                <span class="level-start">
                    <span class="level-item">September 2019</span>
                </span>
                <span class="level-end">
                    <span class="level-item tag">2</span>
                </span>
            </a>
        </li>
        
        </ul>
        </div>
    </div>
</div>
        
            <div class="card widget">
    <div class="card-content">
        <div class="menu">
            <h3 class="menu-label">
                Etiquetas
            </h3>
            <div class="field is-grouped is-grouped-multiline">
                
            </div>
        </div>
    </div>
</div>
        
        </div>
    
</div>

                




<div class="column is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only has-order-3 column-right ">
    
        
<div class="card widget">
    <div class="card-content">
        <h3 class="menu-label">
            Recientes
        </h3>
        
        <article class="media">
            
            <a href="/2019/10/30/复杂基因组组装/" class="media-left">
                <p class="image is-64x64">
                    <img class="thumbnail" src="/images/thumbnail.svg" alt="一个高杂合真菌基因组组装脚本">
                </p>
            </a>
            
            <div class="media-content">
                <div class="content">
                    <div><time class="has-text-grey is-size-7 is-uppercase" datetime="2019-10-30T15:13:32.836Z">2019-10-30</time></div>
                    <a href="/2019/10/30/复杂基因组组装/" class="title has-link-black-ter is-size-6 has-text-weight-normal">一个高杂合真菌基因组组装脚本</a>
                    <p class="is-size-7 is-uppercase">
                        
                    </p>
                </div>
            </div>
        </article>
        
        <article class="media">
            
            <a href="/2019/09/16/GO-annotation/" class="media-left">
                <p class="image is-64x64">
                    <img class="thumbnail" src="/images/thumbnail.svg" alt="De novo 基因组 GO 注释小窗口">
                </p>
            </a>
            
            <div class="media-content">
                <div class="content">
                    <div><time class="has-text-grey is-size-7 is-uppercase" datetime="2019-09-16T12:32:41.938Z">2019-09-16</time></div>
                    <a href="/2019/09/16/GO-annotation/" class="title has-link-black-ter is-size-6 has-text-weight-normal">De novo 基因组 GO 注释小窗口</a>
                    <p class="is-size-7 is-uppercase">
                        
                    </p>
                </div>
            </div>
        </article>
        
        <article class="media">
            
            <a href="/2019/09/16/hello-world/" class="media-left">
                <p class="image is-64x64">
                    <img class="thumbnail" src="/images/thumbnail.svg" alt="Hello World">
                </p>
            </a>
            
            <div class="media-content">
                <div class="content">
                    <div><time class="has-text-grey is-size-7 is-uppercase" datetime="2019-09-16T03:35:12.762Z">2019-09-16</time></div>
                    <a href="/2019/09/16/hello-world/" class="title has-link-black-ter is-size-6 has-text-weight-normal">Hello World</a>
                    <p class="is-size-7 is-uppercase">
                        
                    </p>
                </div>
            </div>
        </article>
        
    </div>
</div>

    
        <div class="card widget">
    <div class="card-content">
        <div class="menu">
        <h3 class="menu-label">
            Archivos
        </h3>
        <ul class="menu-list">
        
        <li>
            <a class="level is-marginless" href="/archives/2019/10/">
                <span class="level-start">
                    <span class="level-item">October 2019</span>
                </span>
                <span class="level-end">
                    <span class="level-item tag">1</span>
                </span>
            </a>
        </li>
        
        <li>
            <a class="level is-marginless" href="/archives/2019/09/">
                <span class="level-start">
                    <span class="level-item">September 2019</span>
                </span>
                <span class="level-end">
                    <span class="level-item tag">2</span>
                </span>
            </a>
        </li>
        
        </ul>
        </div>
    </div>
</div>
    
        <div class="card widget">
    <div class="card-content">
        <div class="menu">
            <h3 class="menu-label">
                Etiquetas
            </h3>
            <div class="field is-grouped is-grouped-multiline">
                
            </div>
        </div>
    </div>
</div>
    
    
</div>

            </div>
        </div>
    </section>
    <footer class="footer">
    <div class="container">
        <div class="level">
            <div class="level-start has-text-centered-mobile">
                <a class="footer-logo is-block has-mb-6" href="/">
                
                    <img src="/images/og_image.png" alt="一个高杂合真菌基因组组装脚本" height="28">
                
                </a>
                <p class="is-size-7">
                &copy; 2019 詹梦涛&nbsp;
                Powered by <a href="https://hexo.io/" target="_blank">Hexo</a> & <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank">Icarus</a>
                
                </p>
            </div>
            <div class="level-end">
            
                <div class="field has-addons is-flex-center-mobile has-mt-5-mobile is-flex-wrap is-flex-middle">
                
                
                <p class="control">
                    <a class="button is-white is-large" target="_blank" title="Creative Commons" href="https://creativecommons.org/">
                        
                        <i class="fab fa-creative-commons"></i>
                        
                    </a>
                </p>
                
                <p class="control">
                    <a class="button is-white is-large" target="_blank" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/">
                        
                        <i class="fab fa-creative-commons-by"></i>
                        
                    </a>
                </p>
                
                <p class="control">
                    <a class="button is-white is-large" target="_blank" title="Download on GitHub" href="https://github.com/ppoffice/hexo-theme-icarus">
                        
                        <i class="fab fa-github"></i>
                        
                    </a>
                </p>
                
                </div>
            
            </div>
        </div>
    </div>
</footer>
    <script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/moment@2.22.2/min/moment-with-locales.min.js"></script>
<script>moment.locale("zh");</script>

<script>
var IcarusThemeSettings = {
    article: {
        highlight: {
            clipboard: true,
            fold: 'unfolded'
        }
    }
};
</script>


    <script src="https://cdn.jsdelivr.net/npm/clipboard@2.0.4/dist/clipboard.min.js" defer></script>



    
    
    
    <script src="/js/animation.js"></script>
    

    
    
    
    <script src="https://cdn.jsdelivr.net/npm/lightgallery@1.6.8/dist/js/lightgallery.min.js" defer></script>
    <script src="https://cdn.jsdelivr.net/npm/justifiedGallery@3.7.0/dist/js/jquery.justifiedGallery.min.js" defer></script>
    <script src="/js/gallery.js" defer></script>
    

    
    

<div id="outdated">
    <h6>Your browser is out-of-date!</h6>
    <p>Update your browser to view this website correctly. <a id="btnUpdateBrowser" href="http://outdatedbrowser.com/">Update
            my browser now </a></p>
    <p class="last"><a href="#" id="btnCloseUpdateBrowser" title="Close">&times;</a></p>
</div>
<script src="https://cdn.jsdelivr.net/npm/outdatedbrowser@1.1.5/outdatedbrowser/outdatedbrowser.min.js" defer></script>
<script>
    document.addEventListener("DOMContentLoaded", function () {
        outdatedBrowser({
            bgColor: '#f25648',
            color: '#ffffff',
            lowerThan: 'flex'
        });
    });
</script>


    
    
<script src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/unpacked/MathJax.js?config=TeX-MML-AM_CHTML" defer></script>
<script>
document.addEventListener('DOMContentLoaded', function () {
    MathJax.Hub.Config({
        'HTML-CSS': {
            matchFontHeight: false
        },
        SVG: {
            matchFontHeight: false
        },
        CommonHTML: {
            matchFontHeight: false
        },
        tex2jax: {
            inlineMath: [
                ['$','$'],
                ['\\(','\\)']
            ]
        }
    });
});
</script>

    
    

<a id="back-to-top" title="Back to Top" href="javascript:;">
    <i class="fas fa-chevron-up"></i>
</a>
<script src="/js/back-to-top.js" defer></script>


    
    

    
    
    
    

    
    
    
    
    


<script src="/js/main.js" defer></script>

    
    <div class="searchbox ins-search">
    <div class="searchbox-container ins-search-container">
        <div class="searchbox-input-wrapper">
            <input type="text" class="searchbox-input ins-search-input" placeholder="Type something...">
            <span class="searchbox-close ins-close ins-selectable"><i class="fa fa-times-circle"></i></span>
        </div>
        <div class="searchbox-result-wrapper ins-section-wrapper">
            <div class="ins-section-container"></div>
        </div>
    </div>
</div>
<script>
    (function (window) {
        var INSIGHT_CONFIG = {
            TRANSLATION: {
                POSTS: 'Entradas',
                PAGES: 'Pages',
                CATEGORIES: 'Categorias',
                TAGS: 'Etiquetas',
                UNTITLED: '(Untitled)',
            },
            CONTENT_URL: '/content.json',
        };
        window.INSIGHT_CONFIG = INSIGHT_CONFIG;
    })(window);
</script>
<script src="/js/insight.js" defer></script>
<link rel="stylesheet" href="/css/search.css">
<link rel="stylesheet" href="/css/insight.css">
    
</body>
</html>